<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<!-- saved from url=(0053)https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/ -->
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <!-- AppResources meta begin -->
        <script type="text/javascript">var ncbi_startTime = new Date();</script>
        <!-- AppResources meta end -->
        
        <!-- TemplateResources meta begin -->
        <meta name="paf_template" content="">

        <!-- TemplateResources meta end -->
        
        <!-- Logger begin -->
        <meta name="ncbi_db" content="pmc"><meta name="ncbi_pdid" content="article"><meta name="ncbi_acc" content=""><meta name="ncbi_domain" content="nihpa"><meta name="ncbi_report" content="record"><meta name="ncbi_type" content="fulltext"><meta name="ncbi_objectid" content=""><meta name="ncbi_pcid" content="/articles/PMC5531759/"><meta name="ncbi_app" content="pmc">
        <!-- Logger end -->
        
        <title>Why are de Bruijn graphs useful for genome assembly?</title>
        
        <!-- AppResources external_resources begin -->
        <link rel="stylesheet" href="./Why are de Bruijn graphs useful for genome assembly__files/jig.min.css"><script type="text/javascript" src="./Why are de Bruijn graphs useful for genome assembly__files/jig.min.js"></script>

        <!-- AppResources external_resources end -->
        
        <!-- Page meta begin -->
        <meta name="robots" content="INDEX,NOFOLLOW,NOARCHIVE"><link rel="canonical" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/"><link rel="schema.DC" href="http://purl.org/DC/elements/1.0/"><meta name="citation_journal_title" content="Nature biotechnology"><meta name="citation_title" content="Why are de Bruijn graphs useful for genome assembly?"><meta name="citation_authors" content="Phillip E. C. Compeau, Pavel A. Pevzner, Glenn Tesler"><meta name="citation_date" content="8 November 2011"><meta name="citation_issue" content="11"><meta name="citation_volume" content="29"><meta name="citation_firstpage" content="987"><meta name="citation_doi" content="10.1038/nbt.2023"><meta name="citation_pmid" content="22068540"><meta name="DC.Title" content="Why are de Bruijn graphs useful for genome assembly?"><meta name="DC.Type" content="Text"><meta name="DC.Publisher" content="NIH Public Access"><meta name="DC.Contributor" content="Phillip E. C. Compeau"><meta name="DC.Contributor" content="Pavel A. Pevzner"><meta name="DC.Contributor" content="Glenn Tesler"><meta name="DC.Date" content="2011 Nov 8"><meta name="DC.Identifier" content="10.1038/nbt.2023"><meta name="DC.Language" content="en"><meta property="og:title" content="Why are de Bruijn graphs useful for genome assembly?"><meta property="og:type" content="article"><meta property="og:url" content="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/"><meta property="og:site_name" content="PubMed Central (PMC)"><meta property="og:image" content="https://www.ncbi.nlm.nih.gov/corehtml/pmc/pmcgifs/pmc-logo-share.png"><meta name="twitter:card" content="summary"><meta name="twitter:site" content="@ncbi"><link rel="stylesheet" href="./Why are de Bruijn graphs useful for genome assembly__files/pmc.min.css" type="text/css"><link rel="stylesheet" href="./Why are de Bruijn graphs useful for genome assembly__files/pmc_extras_prnt.min.css" type="text/css" media="print"><script type="text/javascript" src="./Why are de Bruijn graphs useful for genome assembly__files/common.min.js">//</script><script type="text/javascript" src="./Why are de Bruijn graphs useful for genome assembly__files/NcbiTagServer.min.js">//</script><meta name="citationexporter" content="backend:&#39;/pmc/utils/ctxp/&#39;"><script type="text/javascript" src="./Why are de Bruijn graphs useful for genome assembly__files/jquery.citationexporter.js">//</script><link rel="stylesheet" href="./Why are de Bruijn graphs useful for genome assembly__files/citationexporter.css" type="text/css"><script type="text/javascript" src="./Why are de Bruijn graphs useful for genome assembly__files/MathJax.js"></script><script type="text/javascript">window.name="mainwindow";</script><style type="text/css">.pmc-wm {background:transparent repeat-y top left;background-image:url(/corehtml/pmc/pmcgifs/wm-hhspa.gif);background-size: auto, contain}</style><style type="text/css">.print-view{display:block}</style><style type="text/css">
        div.pmc_para_cit li.highlight,
        div.pmc_para_cit li.highlight .one_line_source
        { background: #E0E0E0; }
        a.bibr.highlight { background: #E0E0E0; } 
      </style><meta name="cited_in_systematic_reviews" content=""><link rel="alternate" type="application/epub+zip" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/epub/"><link rel="alternate" type="application/pdf" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/pdf/nihms329513.pdf">

        <!-- Page meta end -->
    <link rel="shortcut icon" href="https://www.ncbi.nlm.nih.gov/favicon.ico"><meta name="ncbi_phid" content="F4FB7F81B0DB00B100000000005B0054.m_8">
<meta name="referrer" content="origin-when-cross-origin"><link type="text/css" rel="stylesheet" href="./Why are de Bruijn graphs useful for genome assembly__files/4143404.css"><link type="text/css" rel="stylesheet" href="./Why are de Bruijn graphs useful for genome assembly__files/4157116.css" media="print"><style type="text/css">.MathJax_Preview {color: #888}
#MathJax_Message {position: fixed; left: 1em; bottom: 1.5em; background-color: #E6E6E6; border: 1px solid #959595; margin: 0px; padding: 2px 8px; z-index: 102; color: black; font-size: 80%; width: auto; white-space: nowrap}
#MathJax_MSIE_Frame {position: absolute; top: 0; left: 0; width: 0px; z-index: 101; border: 0px; margin: 0px; padding: 0px}
.MathJax_Error {color: #CC0000; font-style: italic}
</style><style type="text/css">.MathJax_Hover_Frame {border-radius: .25em; -webkit-border-radius: .25em; -moz-border-radius: .25em; -khtml-border-radius: .25em; box-shadow: 0px 0px 15px #83A; -webkit-box-shadow: 0px 0px 15px #83A; -moz-box-shadow: 0px 0px 15px #83A; -khtml-box-shadow: 0px 0px 15px #83A; border: 1px solid #A6D ! important; display: inline-block; position: absolute}
.MathJax_Menu_Button .MathJax_Hover_Arrow {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 4px; -webkit-border-radius: 4px; -moz-border-radius: 4px; -khtml-border-radius: 4px; font-family: 'Courier New',Courier; font-size: 9px; color: #F0F0F0}
.MathJax_Menu_Button .MathJax_Hover_Arrow span {display: block; background-color: #AAA; border: 1px solid; border-radius: 3px; line-height: 0; padding: 4px}
.MathJax_Hover_Arrow:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_Hover_Arrow:hover span {background-color: #CCC!important}
</style><style type="text/css">#MathJax_Zoom {position: absolute; background-color: #F0F0F0; overflow: auto; display: block; z-index: 301; padding: .5em; border: 1px solid black; margin: 0; font-weight: normal; font-style: normal; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; -webkit-box-sizing: content-box; -moz-box-sizing: content-box; box-sizing: content-box; box-shadow: 5px 5px 15px #AAAAAA; -webkit-box-shadow: 5px 5px 15px #AAAAAA; -moz-box-shadow: 5px 5px 15px #AAAAAA; -khtml-box-shadow: 5px 5px 15px #AAAAAA; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_ZoomOverlay {position: absolute; left: 0; top: 0; z-index: 300; display: inline-block; width: 100%; height: 100%; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
#MathJax_ZoomFrame {position: relative; display: inline-block; height: 0; width: 0}
#MathJax_ZoomEventTrap {position: absolute; left: 0; top: 0; z-index: 302; display: inline-block; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
</style><style type="text/css">#MathJax_About {position: fixed; left: 50%; width: auto; text-align: center; border: 3px outset; padding: 1em 2em; background-color: #DDDDDD; color: black; cursor: default; font-family: message-box; font-size: 120%; font-style: normal; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; border-radius: 15px; -webkit-border-radius: 15px; -moz-border-radius: 15px; -khtml-border-radius: 15px; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_About.MathJax_MousePost {outline: none}
.MathJax_Menu {position: absolute; background-color: white; color: black; width: auto; padding: 5px 0px; border: 1px solid #CCCCCC; margin: 0; cursor: default; font: menu; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; border-radius: 5px; -webkit-border-radius: 5px; -moz-border-radius: 5px; -khtml-border-radius: 5px; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
.MathJax_MenuItem {padding: 1px 2em; background: transparent}
.MathJax_MenuArrow {position: absolute; right: .5em; padding-top: .25em; color: #666666; font-size: .75em}
.MathJax_MenuActive .MathJax_MenuArrow {color: white}
.MathJax_MenuArrow.RTL {left: .5em; right: auto}
.MathJax_MenuCheck {position: absolute; left: .7em}
.MathJax_MenuCheck.RTL {right: .7em; left: auto}
.MathJax_MenuRadioCheck {position: absolute; left: .7em}
.MathJax_MenuRadioCheck.RTL {right: .7em; left: auto}
.MathJax_MenuLabel {padding: 1px 2em 3px 1.33em; font-style: italic}
.MathJax_MenuRule {border-top: 1px solid #DDDDDD; margin: 4px 3px}
.MathJax_MenuDisabled {color: GrayText}
.MathJax_MenuActive {background-color: #606872; color: white}
.MathJax_MenuDisabled:focus, .MathJax_MenuLabel:focus {background-color: #E8E8E8}
.MathJax_ContextMenu:focus {outline: none}
.MathJax_ContextMenu .MathJax_MenuItem:focus {outline: none}
#MathJax_AboutClose {top: .2em; right: .2em}
.MathJax_Menu .MathJax_MenuClose {top: -10px; left: -10px}
.MathJax_MenuClose {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; font-family: 'Courier New',Courier; font-size: 24px; color: #F0F0F0}
.MathJax_MenuClose span {display: block; background-color: #AAA; border: 1.5px solid; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; line-height: 0; padding: 8px 0 6px}
.MathJax_MenuClose:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_MenuClose:hover span {background-color: #CCC!important}
.MathJax_MenuClose:hover:focus {outline: none}
</style><style type="text/css" media="print">.print-log { position:absolute;left:-10000px;top:auto;width:1px;height:1px;overflow:hidden; }.print-log li { list-style-image: url('https://www.ncbi.nlm.nih.gov/stat?jsevent=print&ncbi_app=pmc&ncbi_db=pmc&ncbi_pcid=%2Farticles%2FPMC5531759%2F&ncbi_pdid=article&ncbi_phid=F4FB7F81B0DB00B100000000005B0054.m_8'); }</style><style type="text/css">.print-log { position:absolute;left:-10000px;top:auto;width:1px;height:1px;overflow:hidden; }</style></head>
    <body class="article" id="ui-ncbiexternallink-3"><div id="MathJax_Message" style="display: none;"></div>
        <div class="grid">
            <div class="col twelve_col nomargin shadow">
                <!-- System messages like service outage or JS required; this is handled by the TemplateResources portlet -->
                <div class="sysmessages">
                    <noscript>
	<p class="nojs">
	<strong>Warning:</strong>
	The NCBI web site requires JavaScript to function. 
	<a href="/guide/browsers/#enablejs" title="Learn how to enable JavaScript" target="_blank">more...</a>
	</p>
	</noscript>
                </div>
                <!--/.sysmessage-->
                <div class="wrap">
                    <div class="page">
                        <div class="top">
                            <div class="universal_header" id="universal_header"><ul class="inline_list jig-ncbimenu ui-ncbimenu resources_list ui-ncbibasicmenu orientation_hort" id="navcontent" role="menubar"><li class="ui-ncbimenu-item-leaf ui-ncbimenu-item-first ui-helper-reset ui-ncbimenu-item-no-hlt" role="presentation"><a class="ui-ncbimenu-link-first" href="https://www.ncbi.nlm.nih.gov/" role="menuitem" title="NCBI Home" id="ncbihome" accesskey="1"><span class="offscreen_noflow">NCBI</span><img src="./Why are de Bruijn graphs useful for genome assembly__files/28977" class="ncbi_logo" title="NCBI" alt="NCBI Logo"></a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#maincontent" title="Skip to the content" tabindex="0" accesskey="3">Skip to main
                        content</a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#navcontent" title="Skip to the navigation" tabindex="0" accesskey="4">Skip to
                        navigation</a></li><li id="resource-menu" class="topmenu ui-helper-reset ui-ncbimenu-item-first ui-helper-reset" role="presentation"><a class="ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown" href="https://www.ncbi.nlm.nih.gov/static/header_footer_ajax/submenu/#resources" tabindex="-1" role="menuitem">Resources</a></li><li id="all-howtos-menu" class="topmenu ui-helper-reset ui-ncbimenu-item-first" role="presentation"><a class="ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown" href="https://www.ncbi.nlm.nih.gov/static/header_footer_ajax/submenu/#howto" tabindex="-1" role="menuitem">How To</a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="https://www.ncbi.nlm.nih.gov/guide/browsers/#accesskeys" title="About My NCBI Accesskeys" tabindex="0" accesskey="0">About NCBI Accesskeys</a></li></ul><div class="myncbi"><span id="myncbiusername" style="display:none"><a href="https://www.ncbi.nlm.nih.gov/account/settings/" id="mnu" title="Edit account settings"></a></span><a accesskey="2" href="https://www.ncbi.nlm.nih.gov/myncbi/" id="myncbi" style="display:none">My NCBI</a><a href="https://www.ncbi.nlm.nih.gov/account/?back_url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC5531759%2F" id="sign_in">Sign in to NCBI</a><a href="https://www.ncbi.nlm.nih.gov/account/signout/?back_url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC5531759%2F" id="sign_out" style="display:none">Sign Out</a></div></div>
                            <div class="header">
    <div class="res_logo">
  <h1 class="img_logo"><a href="https://www.ncbi.nlm.nih.gov/pmc/" class="pmc_logo offscreen">PMC</a></h1>
  <div class="NLMLogo">
    <a href="https://www.nlm.nih.gov/" title="US National Library of Medicine">US National Library of Medicine</a>
    <br>
    <a href="https://www.nih.gov/" title="National Institutes of Health">National Institutes of Health</a>
  </div>
</div>
    <div class="search"><form method="get" action="https://www.ncbi.nlm.nih.gov/pmc/"><div class="search_form"><label for="database" class="offscreen_noflow">Search database</label><select id="database"><optgroup label="Recent"><option value="pmc" selected="selected" class="last" data-ac_dict="pmc-search-autocomplete">PMC</option></optgroup><optgroup label="All"><option value="gquery">All Databases</option><option value="assembly">Assembly</option><option value="biocollections">Biocollections</option><option value="bioproject">BioProject</option><option value="biosample">BioSample</option><option value="biosystems">BioSystems</option><option value="books">Books</option><option value="clinvar">ClinVar</option><option value="clone">Clone</option><option value="cdd">Conserved Domains</option><option value="gap">dbGaP</option><option value="dbvar">dbVar</option><option value="nucest">EST</option><option value="gene">Gene</option><option value="genome">Genome</option><option value="gds">GEO DataSets</option><option value="geoprofiles">GEO Profiles</option><option value="nucgss">GSS</option><option value="gtr">GTR</option><option value="homologene">HomoloGene</option><option value="ipg">Identical Protein Groups</option><option value="medgen">MedGen</option><option value="mesh">MeSH</option><option value="ncbisearch">NCBI Web Site</option><option value="nlmcatalog">NLM Catalog</option><option value="nuccore">Nucleotide</option><option value="omim">OMIM</option><option value="pmc" data-ac_dict="pmc-search-autocomplete">PMC</option><option value="popset">PopSet</option><option value="probe">Probe</option><option value="protein">Protein</option><option value="proteinclusters">Protein Clusters</option><option value="pcassay">PubChem BioAssay</option><option value="pccompound">PubChem Compound</option><option value="pcsubstance">PubChem Substance</option><option value="pubmed">PubMed</option><option value="pubmedhealth">PubMed Health</option><option value="snp">SNP</option><option value="sparcle">Sparcle</option><option value="sra">SRA</option><option value="structure">Structure</option><option value="taxonomy">Taxonomy</option><option value="toolkit">ToolKit</option><option value="toolkitall">ToolKitAll</option><option value="toolkitbookgh">ToolKitBookgh</option><option value="unigene">UniGene</option></optgroup></select><div class="nowrap"><label for="term" class="offscreen_noflow" accesskey="/">Search term</label><div class="nowrap"><div class="jig-ncbiclearbutton-wrap ui-ncbiclearbutton-wrap box-shadow"><input type="text" name="term" id="term" title="Search PMC. Use up and down arrows to choose an item from the autocomplete." value="" class="jig-ncbiclearbutton jig-ncbiautocomplete" data-jigconfig="dictionary:&#39;pmc-search-autocomplete&#39;,disableUrl:&#39;NcbiSearchBarAutoComplCtrl&#39;" autocomplete="off" data-sbconfig="ds:&#39;no&#39;,pjs:&#39;no&#39;,afs:&#39;yes&#39;" aria-haspopup="true" aria-autocomplete="list" role="textbox"><a class="reset" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" style="visibility: hidden;"><img src="./Why are de Bruijn graphs useful for genome assembly__files/clear.png" alt="Clear input"></a></div></div><button id="search" type="submit" class="button_search nowrap" cmd="go">Search</button></div></div></form><ul class="searchlinks inline_list"><li>
                        <a href="https://www.ncbi.nlm.nih.gov/pmc/advanced/">Advanced</a>
                    </li><li>
                        <a href="https://www.ncbi.nlm.nih.gov/pmc/journals/">Journal list</a>
                    </li><li class="help">
                        <a target="_blank" href="https://www.ncbi.nlm.nih.gov/books/NBK3825/">Help</a>
                    </li></ul></div>
</div>

                            
                            
                        <!--<component id="Page" label="headcontent"/>-->
                            
                        </div>
                        <div class="content">
                            <!-- site messages -->
                            <div class="container">
    <div id="maincontent" class="content eight_col col">
        <div class="navlink-box">
            <ul class="page-breadcrumbs inline_list small"><li class="journal-list"><a href="https://www.ncbi.nlm.nih.gov/pmc/journals/" class="navlink">Journal List</a></li><li class="article-entrez-filter"><a href="https://www.ncbi.nlm.nih.gov/pmc/?term=hhs%20author%20manuscript[filter]" class="navlink">HHS Author Manuscripts</a></li><li class="accid">PMC5531759</li></ul>
        </div>

        <!-- Journal banner -->
        <div class="pmc-page-banner whole_rhythm"><div><img src="./Why are de Bruijn graphs useful for genome assembly__files/logo-hhspa.png" alt="Logo of nihpa" usemap="#logo-imagemap"><map id="logo-imagemap" name="logo-imagemap"><area shape="rect" coords="0,57,255,75" alt="About Author manuscripts" title="About Author manuscripts" href="https://www.ncbi.nlm.nih.gov/pmc/about/authorms.html" ref="https://www.ncbi.nlm.nih.gov/pmc/about/authorms.html"><area shape="rect" coords="256,57,500,75" alt="Submit a manuscript" title="Submit a manuscript" href="https://www.nihms.nih.gov/" ref="reftype=publisher&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A&amp;rendering-type=normal" target="pmc_ext"><area shape="rect" coords="0,0,499,74" alt="HHS Public Access; Author Manuscript; Accepted for publication in peer reviewed journal;" title="HHS Public Access; Author Manuscript; Accepted for publication in peer reviewed journal;" href="https://www.ncbi.nlm.nih.gov/pmc/about/public-access/" ref="reftype=publisher&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A&amp;rendering-type=normal" target="pmc_ext"></map></div> </div>
        
        <!--component id='MainPortlet' label='search-reference'/-->
        
        <!-- Book content -->
        <div class="">
            
        
            
            <div class="hide-overflow article lit-style content pmc-wm slang-all page-box"><!--main-content--><div class="jig-ncbiinpagenav" data-jigconfig="smoothScroll: false, allHeadingLevels: [&#39;h2&#39;], headingExclude: &#39;:hidden&#39;" id="ui-ncbiinpagenav-1"><div class="fm-sec half_rhythm no_top_margin"><div class="fm-citation half_rhythm no_top_margin clearfix"><div class="small"><div class="inline_block eight_col va_top"><div><span role="menubar"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" role="menuitem" aria-expanded="false" aria-haspopup="true">Nat Biotechnol</a></span>. Author manuscript; available in PMC 2017 Jul 27.</div><div></div><div>Published in final edited form as:</div><div style="margin-left:1em"><div class="fm-vol-iss-date"><a href="https://www.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;retmode=ref&amp;cmd=prlinks&amp;id=22068540" target="pmc_ext" ref="reftype=publisher&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CFront%20Matter&amp;TO=Content%20Provider%7CArticle%7CRestricted%20Access&amp;rendering-type=normal"><span class="cit">Nat Biotechnol. 2011 Nov 8; 29(11): 987–991. </span></a></div><span class="fm-vol-iss-date">Published online 2011 Nov 8. </span> <span class="doi">doi:&nbsp; <a href="https://dx.doi.org/10.1038%2Fnbt.2023" target="pmc_ext" ref="reftype=other&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CFront%20Matter&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">10.1038/nbt.2023</a></span></div></div><div class="inline_block four_col va_top show-overflow align_right"><div class="fm-citation-ids"><div class="fm-citation-pmcid"><span class="fm-citation-ids-label">PMCID: </span><span>PMC5531759</span></div><div class="fm-citation-manuscriptid"><span class="fm-citation-ids-label">NIHMSID: </span><span>NIHMS329513</span></div><div class="fm-citation-pmid">PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22068540">22068540</a></div></div></div></div></div><h1 class="content-title">Why are de Bruijn graphs useful for genome assembly?</h1><div class="half_rhythm"><div class="contrib-group fm-author"><a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=Compeau%20PE%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=22068540" class="affpopup" co-rid="_co_idm140318147259408" co-class="co-affbox">Phillip E. C. Compeau</a>, <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=Pevzner%20PA%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=22068540" class="affpopup" co-rid="_co_idm140318146380656" co-class="co-affbox">Pavel A. Pevzner</a>, and  <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=Tesler%20G%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=22068540" class="affpopup" co-rid="_co_idm140318151135248" co-class="co-affbox">Glenn Tesler</a></div><div style="display:none" class="contrib-group aff-tip"><div id="_co_idm140318147259408"><h3 class="no_margin">Phillip E. C. Compeau</h3><p>Department of Mathematics, University of California San Diego, La Jolla, CA, USA</p><div>Find articles by <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=Compeau%20PE%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=22068540">Phillip E. C. Compeau</a></div></div><div id="_co_idm140318146380656"><h3 class="no_margin">Pavel A. Pevzner</h3><p>Department of Computer Science and Engineering, University of California San Diego, La Jolla, CA, USA</p><div>Find articles by <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=Pevzner%20PA%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=22068540">Pavel A. Pevzner</a></div></div><div id="_co_idm140318151135248"><h3 class="no_margin">Glenn Tesler</h3><p>Department of Mathematics, University of California San Diego, La Jolla, CA, USA</p><div>Find articles by <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=Tesler%20G%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=22068540">Glenn Tesler</a></div></div></div></div><div class="fm-panel small half_rhythm"><div class="fm-authors-info fm-panel hide half_rhythm" id="idm140318146057888_ai" style="display:none"><div><span class="fm-affl">Phillip E. C. Compeau, </span><span class="fm-affl" lang="en" id="A1"> Department of Mathematics, University of California San Diego, La Jolla, CA, USA;</span></div><div><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#idm140318146057888aff-info">Contributor Information</a>.</div></div><div class="togglers"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" class="pmctoggle" rid="idm140318146057888_ai">Author information <span>►</span></a> <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" class="pmctoggle" rid="idm140318146057888_cpl">Copyright and License information <span>►</span></a> <a href="https://www.ncbi.nlm.nih.gov/pmc/about/disclaimer/">Disclaimer</a></div><div class="fm-article-notes fm-panel half_rhythm"></div><div class="fm-cpl-info fm-panel hide half_rhythm" id="idm140318146057888_cpl" style="display:none"><div class="fm-copyright half_rhythm"><a href="https://www.ncbi.nlm.nih.gov/pmc/about/copyright/">Copyright notice</a> </div><div class="fm-copyright half_rhythm"><div class="license">Users may view, print, copy, download and text and data- mine the content in such documents, for the purposes of academic research, subject always to the full Conditions of use:
<a href="http://www.nature.com/authors/editorial_policies/license.html#terms">http://www.nature.com/authors/editorial_policies/license.html#terms</a></div></div></div></div><div id="pmclinksbox" class="links-box whole_rhythm"><div class="small"><div>The publisher's final edited version of this article is available  at <a href="https://www.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;retmode=ref&amp;cmd=prlinks&amp;id=22068540" target="pmc_ext" ref="reftype=publisher&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CFront%20Matter&amp;TO=Content%20Provider%7CArticle%7CRestricted%20Access&amp;rendering-type=normal">Nat Biotechnol</a></div><div>See other articles in PMC that <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/citedby/">cite</a> the published article.</div></div></div></div><div class="sec"></div><div id="S1" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><span role="menubar"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" title="Go to other sections in this page" role="menuitem" aria-expanded="false" aria-haspopup="true">Go to:</a></span></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="S1title">Assembling billions of short sequencing reads into a contiguous genome is a formidable challenge</h2><p id="P1" class="p p-first">The development of algorithmic ideas for Next-Generation Sequencing (NGS) can be traced back three hundred years to the Prussian city of Königsberg (present-day Kaliningrad, Russia), where seven bridges joined the four parts of the city located on opposing banks of the Pregel River and two river islands (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F1/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="F1" rid-ob="ob-F1" co-legend-rid="lgnd_F1"><span>Fig. 1a</span></a>). Königsberg’s residents enjoyed strolling through the city, and they wondered: is it possible to visit every part of the city by walking across each of the seven bridges exactly once and returning to one’s starting location? Remarkably, the conceptual breakthrough used in 1735 to solve this <em>Bridges of Königsberg Problem</em> by the great mathematician Leonhard Euler<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R1" rid="R1" class=" bibr popnode" role="button" aria-expanded="false" aria-haspopup="true">1</a></sup> also enables the assembly of billions of short sequencing reads.</p><!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock whole_rhythm clearfix" id="F1" co-legend-rid="lgnd_F1"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F1/" target="figure" rid-figpopup="F1" rid-ob="ob-F1"><!--fig/graphic|fig/alternatives/graphic mode="anchored" m1--></a><div data-largeobj="" data-largeobj-link-rid="largeobj_idm140318147276528" class="figure"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F1/" target="figure" rid-figpopup="F1" rid-ob="ob-F1"></a><a class="inline_block ts_canvas" href="https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click%20on%20image%20to%20zoom&amp;p=PMC3&amp;id=5531759_nihms329513f1.jpg" target="tileshopwindow"><div class="ts_bar small" title="Click on image to zoom"></div><img alt="An external file that holds a picture, illustration, etc.
Object name is nihms329513f1.jpg" title="Click on image to zoom" class="tileshop" src="./Why are de Bruijn graphs useful for genome assembly__files/nihms329513f1.jpg"></a></div><div id="largeobj_idm140318147276528" class="largeobj-link align_right" style="display: none"><a target="object" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F1/?report=objectonly">Open in a separate window</a></div><div class="icnblk_cntnt" id="lgnd_F1"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F1/" target="figure" rid-figpopup="F1" rid-ob="ob-F1">Figure 1</a></div><!--caption a7--><div class="caption"><!--caption a8--><strong>Bridges of Konigsberg problem</strong><p id="P19">(<strong>a</strong>) A map of old Königsberg, in which each area of the city is labeled with a differently colored point. (<strong>b</strong>) The <em>Königsberg Bridge Graph</em>, formed by representing each of four land areas as a node and each of the city’s seven bridges as an edge.</p></div></div></div><p id="P2" class="p p-last">Euler’s first insight was to represent each landmass as a point (called a node) and each bridge as a line segment (called an edge) connecting the appropriate two points. This creates a graph—a network of nodes connected by edges (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F1/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="F1" rid-ob="ob-F1" co-legend-rid="lgnd_F1"><span>Fig. 1b</span></a>). By describing a procedure for determining whether an arbitrary graph contains an <em>Eulerian</em>
<em>cycle</em> (a path through the graph that visits every edge exactly once and returns back where it started), Euler not only resolved the Bridges of Königsberg Problem but also effectively launched the entire branch of mathematics known today as <em>graph theory</em><sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R2" rid="R2" class=" bibr popnode" role="button" aria-expanded="false" aria-haspopup="true">2</a></sup>.</p></div><div id="S2" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="S2title">Computational issues arise from alignment-based assembly</h2><p id="P3" class="p p-first">To illustrate why graphs are useful for genome assembly, we will use a simple example with five very short reads (CGTGCAA, ATGGCGT, CAATGGC, GGCGTGC and TGCAATG) sequenced from a small circular genome, ATGGCGTGCA (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F2/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="F2" rid-ob="ob-F2" co-legend-rid="lgnd_F2"><span>Fig. 2a</span></a>). Current NGS methods produce reads that vary in length, but the most popular technology today generates approximately 100-nucleotide reads. A straightforward method for assembling reads into longer contiguous sequences—and the one used for assembling the human genome<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R3" rid="R3" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619105" role="button" aria-expanded="false" aria-haspopup="true">3</a>,<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R4" rid="R4" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619101" role="button" aria-expanded="false" aria-haspopup="true">4</a></sup> in 2001 as well as for all other projects based on Sanger sequencing—uses a graph in which each read is represented by a node and overlap between reads is represented by an arrow (called a <em>directed edge</em>) joining two reads. For instance, two nodes representing reads may be connected with a directed edge if the reads overlap by at least five nucleotides (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F2/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="F2" rid-ob="ob-F2" co-legend-rid="lgnd_F2"><span>Fig. 2b</span></a>).</p><!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock whole_rhythm clearfix" id="F2" co-legend-rid="lgnd_F2"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F2/" target="figure" rid-figpopup="F2" rid-ob="ob-F2"><!--fig/graphic|fig/alternatives/graphic mode="anchored" m1--></a><div data-largeobj="" data-largeobj-link-rid="largeobj_idm140318150406896" class="figure"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F2/" target="figure" rid-figpopup="F2" rid-ob="ob-F2"></a><a class="inline_block ts_canvas" href="https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click%20on%20image%20to%20zoom&amp;p=PMC3&amp;id=5531759_nihms329513f2.jpg" target="tileshopwindow"><div class="ts_bar small" title="Click on image to zoom"></div><img alt="An external file that holds a picture, illustration, etc.
Object name is nihms329513f2.jpg" title="Click on image to zoom" class="tileshop" src="./Why are de Bruijn graphs useful for genome assembly__files/nihms329513f2.jpg"></a></div><div id="largeobj_idm140318150406896" class="largeobj-link align_right" style="display: none"><a target="object" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F2/?report=objectonly">Open in a separate window</a></div><div class="icnblk_cntnt" id="lgnd_F2"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F2/" target="figure" rid-figpopup="F2" rid-ob="ob-F2">Figure 2</a></div><!--caption a7--><div class="caption"><!--caption a8--><strong>Two strategies for genome assembly: from Hamiltonian cycles to Eulerian cycles</strong><p id="P20">(<strong>a</strong>) A simplified example of a small circular genome. (<strong>b</strong>) In traditional Sanger sequencing algorithms, reads were represented as nodes in a graph, and edges represented alignments between reads. Walking along a Hamiltonian cycle by following the edges in numerical order allows one to reconstruct the circular genome by combining alignments between successive reads. At the end of the cycle, the sequence wraps around to the start of the genome; the repeated part of the sequence is grayed out in the alignment diagram. (<strong>c</strong>) An alternative assembly technique first splits reads into all possible <em>k</em>-mers: with <em>k</em> = 3, “ATGGCGT” comprises ATG, TGG, GGC, GCG and CGT. Following a Hamiltonian cycle (indicated by red edges) allows one to reconstruct the genome by forming an alignment in which each successive <em>k</em>-mer (from successive nodes) is shifted by one position. This procedure recovers the genome but does not scale well to large graphs. (<strong>d</strong>) Modern short-read assembly algorithms construct a de Bruijn graph by representing all <em>k</em>-mer prefixes and suffixes as nodes, then drawing edges that represent <em>k</em>-mers having a particular prefix and suffix. For example, <em>k</em>-mer edge ATG has prefix AT and suffix TG. Finding an Eulerian cycle allows one to reconstruct the genome by forming an alignment in which each successive <em>k</em>-mer (from successive edges) is shifted by one position. This generates the same cyclic genome sequence without the computational strain of finding a Hamiltonian cycle.</p></div></div></div><p id="P4">Visualizing an ant walking along the edges of this graph provides a useful illustrative aid for understanding a broad class of algorithms used to derive insights from graphs. In the case of genome assembly, the ant’s path traces a series of overlapping reads, and thus represents a candidate assembly. Specifically, if the ant follows the path <strong><em>AT</em></strong>GGCGT → <strong><em>GG</em></strong>CGTGC → <strong><em>CG</em></strong>TGCAA → <strong><em>TG</em></strong>CAATG → <strong><em>CA</em></strong>ATGGC → <strong><em>AT</em></strong>GGCGT, its walk induces a <em>Hamiltonian cycle</em> in our graph, which is a cycle that travels to every node exactly once (but closes with the starting node), meaning that each read will be included once in the assembly. The circular “genome” ATGGCGTGCA resulting from a Hamiltonian cycle contains all five reads and thus reconstructs the original genome (although we may have to “wrap around” the genome, for example in order to locate CAATGGC in <strong>ATGGC</strong>GTG<strong>CA</strong>).</p><p id="P5">Modern assemblers usually work with strings of a particular length <em>k</em> (<em>k-mers</em>), which are shorter than entire reads (see <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#BX2" rid="BX2" class=" boxed-text">Box 2</a> for an explanation of why researchers prefer <em>k</em>-mers to reads). For example, a 100-nucleotide read may be divided into 46 overlapping 55-mers. We can generalize the Hamiltonian Cycle approach to <em>k</em>-mers by constructing a graph as follows.</p><div class="boxed-text-box whole_rhythm hide-overflow" id="BX2"><h3>Box 2</h3><div class="caption"><h4>Practical strategies for applying de Bruijn graphs</h4></div><p id="P26">We have noted that in practice, attempting to apply de Bruijn graphs to experimental data is not a straightforward procedure. Below we describe some key computational techniques that have been devised in order to address the practical challenges introduced by errors and quirks in current sequencing technologies, as well as to resolve the complexities created by repeat-rich genomes.</p><p id="P27">For instance, the astute observer will have noticed that the de Bruijn method for fragment assembly relies upon four hidden assumptions that do not hold for NGS. We took for granted that we can generate all <em>k</em>-mers present in the genome, that all <em>k</em>-mers are error-free, that each <em>k</em>-mer appears at most once in the genome, and that the genome consists of a single circular chromosome. For example, Illumina technology, which generates 100-nucleotide long reads, may miss some 100-mers present in the genome (even if the read coverage is high), and the 100-mers that it does generate typically have errors.</p><div class="sec" id="S7"><h4>Generating (nearly) all <em>k</em>-mers present in the genome</h4><p id="P28" class="p p-first-last">100-mer reads generated by Illumina technology capture only a small fraction of 100-mers from the genome (even for high-coverage NGS projects), thus violating the key assumption of the de Bruijn graphs. However, if one breaks these reads into shorter <em>k</em>-mers, the resulting <em>k</em>-mers often represent nearly all <em>k</em>-mers from the genome for sufficiently small <em>k</em>. For example, NGS assemblers may break every 100-nucleotide read into 46 overlapping 55-mers and further assemble the resulting 55-mers using de Bruijn graphs. Even if some 100-mers occurring in the genome are not generated as reads, this <em>read breaking</em> procedure<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R17" rid="R17" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619113" role="button" aria-expanded="false" aria-haspopup="true">17</a></sup> ensures that nearly all 55-mers appearing in the genome are detected. In our recurring example, our five reads do not account for all 7-mer substrings of the genome, but they do contain all 3-mers present in the genome; we have seen that this is sufficient to reconstruct the genome.</p></div><div class="sec" id="S8"><h4>Handling errors in reads</h4><p id="P29" class="p p-first-last">Each error in a read creates a <em>bulge</em> in the de Bruijn graph (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#SD1" rid="SD1" class=" supplementary-material">Supplementary Fig. 1</a>), complicating assembly. To make matters even worse, in a genome with inexact repeats (such as two regions differing by a single nucleotide or other small variation), reads from the two repeat copies will also generate bulges in the de Bruijn graph. An approach for <em>error-correcting</em> reads, to resolve errors before even beginning assembly, was proposed<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R18" rid="R18" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628838303" role="button" aria-expanded="false" aria-haspopup="true">18</a></sup> in 2001 and such a phase is now common to do before assembly. An approach for removing bulges from de Bruijn graphs was outlined<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R19" rid="R19" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619108" role="button" aria-expanded="false" aria-haspopup="true">19</a></sup> in 2004 and, with some variations, is used in most existing NGS assemblers (EULER-SR<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R20" rid="R20" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619099" role="button" aria-expanded="false" aria-haspopup="true">20</a></sup>, Velvet<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R21" rid="R21" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619109" role="button" aria-expanded="false" aria-haspopup="true">21</a></sup>, ALLPATHS<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R22" rid="R22" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619104" role="button" aria-expanded="false" aria-haspopup="true">22</a></sup>, ABySS<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R23" rid="R23" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619096" role="button" aria-expanded="false" aria-haspopup="true">23</a></sup>, SOAPdenovo<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R24" rid="R24" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619106" role="button" aria-expanded="false" aria-haspopup="true">24</a></sup>). These and other recently developed tools introduced many new algorithmic and software engineering ideas in assembly algorithms and paved the way towards assembling large (e.g., mammalian) genomes with NGS data. See refs. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R25" rid="R25" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619103" role="button" aria-expanded="false" aria-haspopup="true">25</a><sup>–</sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R26" rid="R26" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619110" role="button" aria-expanded="false" aria-haspopup="true">26</a> for a detailed comparison of these and other NGS assemblers.</p></div><div class="sec" id="S9"><h4>Handling DNA repeats</h4><p id="P30" class="p p-first">Imagine for a moment that our (cyclic) genome is ATGCATGC. Then we would hopefully detect four 3-mers: ATG, TGC, GCA, and CAT; however, the present definition of de Bruijn graphs would lead us to reconstruct the genome as ATGC. As one can see, the problem is that each of the 3-mers detected actually occurs <em>twice</em> in the original genome. Therefore, we will need to adjust genome reconstruction so that we not only find all <em>k</em>-mers occurring in the genome, but we also find how many times each such <em>k</em>-mer appears, which is called its <em>k-mer multiplicity.</em> The good news is that we can still handle fragment assembly in the case when <em>k</em>-mer multiplicities are known.</p><p id="P31">We simply use the same method to construct the de Bruijn graph <em>E</em>, except that if the multiplicity of a <em>k</em>-mer is <em>m</em>, we will connect its prefix to its suffix via <em>m</em> directed edges (instead of just one). Extending our ongoing example, if during read generation we discover that each of the four 3-mers TGC, GCG, CGT, and GTG has multiplicity 2, and that each of the six 3-mers ATG, TGG, GGC, GCA, CAA, and AAT has multiplicity 1, we create the graph shown in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#SD1" rid="SD1" class=" supplementary-material">Supplementary Figure 2</a>. Furthermore, it is easy to see that the graph resulting from adding multiplicity edges is Eulerian, as both the indegree and outdegree of a node (representing a (<em>k</em> − 1)-mer) equals the number of times this (<em>k</em> − 1)-mer appears in the genome.</p><p id="P32" class="p p-last">In practice, information about the multiplicities of <em>k</em>-mers in the genome may be difficult to obtain with NGS technologies. However, computer scientists have found ways to reconstruct the genome even when these data are unavailable. One such technique involves <em>paired reads</em>. Reads are typically generated in pairs by sequencing both ends of a long fragment of DNA whose length can be estimated well. If one read maps at or before the entrance to a repeat in the graph, and the other maps at or after the exit, the read pair may be used to determine the correct traversal through the graph.</p></div><div class="sec" id="S10"><h4>Handling multiple and linear chromosomes</h4><p id="P33" class="p p-first-last">Our presentation focused on a genome consisting of one circular chromosome. If our chromosome is linear, then we will instead need to search for an Eulerian <em>path</em>, which is not required to end at the node where it begins; if there are multiple linear chromosomes, then we will have one path for each chromosome. Fortunately, Euler’s work can be adapted to handle these complexities.</p></div><div class="sec" id="S11"><h4>Handling unsequenced regions</h4><p id="P34" class="p p-first">Unsequenced regions and sequencing errors may further break the chromosomes into <em>contigs</em> (a sequenced contiguous region of DNA) and <em>gaps</em> (unsequenced regions), with one path for each contig. Increasing the value of <em>k</em> will tend to reduce the number of bulges and give longer contigs in places with high coverage but some errors. However, it will also tend to break contigs in regions that have low coverage. Successive contigs along a chromosome may have overlaps of fewer than <em>k</em> nucleotides, or may have gaps between them.</p><p id="P35" class="p p-last">The correct order and orientation of the contigs, and the approximate sizes of the gaps, is determined in the <em>scaffolding</em> phase of assembly. This uses additional information, including paired reads, to determine the order of contigs.</p></div></div><p id="P6">First, from a set of reads, form a node for every <em>k</em>-mer appearing in these reads. Second, given a <em>k</em>-mer, define its <em>prefix</em> as the string formed by all its nucleotides except the final one and its <em>suffix</em> as the string formed by all its nucleotides except the first one. Connect one <em>k</em>-mer to another with a directed edge if the suffix of the former equals the prefix of the latter—that is, if the two <em>k</em>-mers completely overlap except for one nucleotide at each end (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F2/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="F2" rid-ob="ob-F2" co-legend-rid="lgnd_F2"><span>Fig. 2c</span></a>). Third, look for a Hamiltonian cycle, which represents a candidate genome because it visits each detected <em>k</em>-mer; moreover, that path will also have minimal length because a Hamiltonian cycle travels to each <em>k</em>-mer exactly once.</p><p id="P7">However, this method is not as easy to implement as it might seem. Imagine attempting to create a similar graph for a single run of an Illumina sequencer that generates many reads. A million (10<sup>6</sup>) reads will require a trillion (10<sup>12</sup>) pairwise alignments. A billion (10<sup>9</sup>) reads necessitate a quintillion (10<sup>18</sup>) alignments. What’s more, there is no known efficient algorithm for finding a Hamiltonian cycle in a large graph with millions (let alone billions) of nodes. The Hamiltonian cycle approach<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R5" rid="R5" class=" bibr popnode" role="button" aria-expanded="false" aria-haspopup="true">5</a>,<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R6" rid="R6" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619102" role="button" aria-expanded="false" aria-haspopup="true">6</a></sup> was feasible for sequencing the first microbial genome<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R7" rid="R7" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619114" role="button" aria-expanded="false" aria-haspopup="true">7</a></sup> in 1995 and the human genome in 2001, as well as for all other projects based on Sanger sequencing. However, the computational burden was so large that most NGS sequencing projects have abandoned the Hamiltonian cycle approach.</p><p id="P8" class="p p-last">And here is where genome sequencing faces the limits of modern computer science: the computational problem of finding a Hamiltonian cycle belongs to a class of problems that are collectively called <em>NP</em>-Complete (see ref. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R2" rid="R2" class=" bibr popnode" role="button" aria-expanded="false" aria-haspopup="true">2</a> for further background). To this day, some of the world’s top computer scientists have worked to find an efficient solution to any <em>NP</em>-Complete problem, with no success. What makes their failure doubly frustrating is that neither has anyone been able to prove that <em>NP</em>-Complete problems are intractable; efficient solutions to these problems may actually exist, but such solutions have not yet been discovered.</p></div><div id="S3" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="S3title">Scalable assembly with de Bruijn graphs</h2><p id="P9" class="p p-first">We have observed that finding a cycle visiting all <em>nodes</em> of a graph exactly once (called the <em>Hamiltonian cycle problem</em>) is a difficult computational problem; however, as we will soon see, finding a cycle visiting all <em>edges</em> of a graph exactly once is much easier. This algorithmic contrast has motivated computer scientists to cast fragment assembly as such a problem. So instead of assigning each <em>k</em>-mer to a node, we will now assign each <em>k</em>-mer located within a read to an edge. This allows the construction of a <em>de Bruijn graph</em>, which we call <em>E</em>, as follows. First, form a node for every distinct prefix or suffix of a <em>k</em>-mer, meaning that a given sequence of length <em>k</em> − 1 can appear only once as a node of the graph. Then, connect node <em>x</em> to node <em>y</em> with a directed edge if some <em>k</em>-mer has prefix <em>x</em> and suffix <em>y</em>, and label the edge with this <em>k</em>-mer (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F2/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="F2" rid-ob="ob-F2" co-legend-rid="lgnd_F2"><span>Fig. 2d</span></a>). For a discussion on the origin of de Bruijn graphs, see <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#BX1" rid="BX1" class=" boxed-text">Box 1</a>.</p><div class="boxed-text-box whole_rhythm hide-overflow" id="BX1"><h3>Box 1</h3><div class="caption"><h4>Historical notes</h4></div><div class="sec" id="S5"><h4>Origin of de Bruijn graphs</h4><p id="P22" class="p p-first">In 1946, the Dutch mathematician Nicolaas de Bruijn became interested in the <em>Superstring Problem</em><sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R12" rid="R12" class=" bibr popnode" role="button" aria-expanded="false" aria-haspopup="true">12</a></sup>: find a shortest circular <em>superstring</em> that contains all possible <em>substrings</em> of length <em>k</em> (<em>k</em>-mers) in a given alphabet. There exist <em>n<sup>k</sup> k</em>-mers in an alphabet containing <em>n</em> symbols: for example, given the alphabet {A,T,G,C}, there are 4<sup>3</sup> = 64 trinucleotides. If our alphabet is instead {0,1}, then all possible 3-mers are simply given by all 3-digit binary numbers: 000, 001, 010, 011, 100, 101, 110, 111. The circular superstring “0001110100” not only contains all 3-mers but is also as short as possible, since it contains each 3-mer exactly once. But how can one <em>construct</em> such a superstring for all <em>k</em>-mers in the case of an arbitrary <em>k</em> and an arbitrary alphabet? De Bruijn answered this question by borrowing Euler’s solution of the Bridges of Königsberg Problem.</p><p id="P23" class="p p-last">So construct a graph <em>B</em> (the original graph called the <em>de Bruijn graph</em>) for which every possible (<em>k</em> − 1)-mer is assigned to a node; connect one (<em>k</em> − 1)-mer by a directed edge to a second (<em>k</em> − 1)-mer if there is some <em>k</em>-mer whose prefix is the former and whose suffix is the latter (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F3/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="F3" rid-ob="ob-F3" co-legend-rid="lgnd_F3"><span>Fig. 3</span></a>). Edges of the de Bruijn graph represent all possible <em>k</em>-mers, and thus an Eulerian cycle in <em>B</em> represents a shortest (cyclic) superstring that contains each <em>k</em>-mer exactly once. By checking that the indegree and outdegree of every node in <em>B</em> equals the size of the alphabet, we can verify that <em>B</em> contains an Eulerian cycle. In turn, we can construct an Eulerian cycle using Euler’s algorithm, therefore solving the superstring problem. It should now be apparent why the “de Bruijn graph” construction described in the main text, which does not use all possible <em>k</em>-mers as edges but rather only those generated from our reads, is also named in honor of de Bruijn.</p></div><div class="sec" id="S6"><h4>Sequencing by Hybridization</h4><p id="P24" class="p p-first">Few people remember that the idea of DNA sequencing using short reads (like in NGS) goes back to late 1980s. In fact, the very first short read sequencing technology, called Sequencing by Hybridization (or SBH), aimed at exactly this goal. SBH<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R13" rid="R13" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619097" role="button" aria-expanded="false" aria-haspopup="true">13</a>–<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R15" rid="R15" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628838302" role="button" aria-expanded="false" aria-haspopup="true">15</a></sup> (proposed in 1988) is based on building a microarray containing every possible oligonucleotide of length <em>k.</em> After hybridization of such an array with an unknown genome, one would get information about all <em>k</em>-mers present in the genome. De Bruijn graphs were first brought to bioinformatics in 1989 as a method to assemble <em>k</em>-mers generated by SBH<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R16" rid="R16" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619112" role="button" aria-expanded="false" aria-haspopup="true">16</a></sup>; this method is very similar to the key algorithmic step of NGS assemblers today.</p><p id="P25" class="p p-last">DNA arrays ultimately failed to realize the dream (DNA sequencing) that motivated their inventors because the fidelity of DNA hybridization with the array turned out to be too low and the value of <em>k</em> was too small. Yet the failure of DNA arrays was short-lived: while the original goal (DNA sequencing) was still out of reach, two new unexpected applications emerged: measuring gene expression and analyzing genetic variations. Today these applications have become so ubiquitous that most people have forgotten that the original goal of the inventors of DNA arrays was to sequence the human genome!</p></div></div><p id="P10">Now imagine an ant that follows a different strategy: instead of visiting every node of the graph (as before), it now visits every <em>edge</em> of <em>E</em> exactly once. Sound familiar? This is exactly the kind of path that would solve the Bridges of Königsberg Problem and is called an <em>Eulerian cycle</em>. Since it visits all edges of <em>E</em>, which represent all possible <em>k</em>-mers, this new ant also spells out a candidate genome: for each edge that the ant traverses, one tacks on the first nucleotide of the <em>k</em>-mer assigned to that edge.</p><p id="P11">Euler considered graphs for which there exists a path between every two nodes (called connected graphs). He proved that a connected graph with undirected edges contains an Eulerian cycle exactly when every node in the graph has an even number of edges touching it. For the Königsberg Bridge Graph, this is not the case because each of the four nodes has an odd number of edges touching it (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F1/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="F1" rid-ob="ob-F1" co-legend-rid="lgnd_F1"><span>Fig. 1b</span></a>), and so the desired stroll through the city does not exist.</p><p id="P12">The case of directed graphs (i.e. graphs with directed edges) is similar. For any node <em>v</em> in a directed graph, define the <em>indegree</em> of <em>v</em> as the number of edges leading into <em>v</em> and the <em>outdegree</em> of <em>v</em> as the number of edges leaving <em>v</em>. A graph in which indegrees are equal to outdegrees for all nodes is called <em>balanced</em>. Euler’s theorem states that a connected directed graph has an Eulerian cycle if and only if it is balanced. In particular, Euler’s theorem implies that the graph <em>E</em> contains an Eulerian cycle as long as we have located all <em>k</em>-mers present in the genome. Indeed, in this case, for any node, both its indegree and outdegree represent the number of times the (<em>k</em> − 1)-mer assigned to that node occurs in the genome.</p><p id="P13">It is easy to see that a graph possessing an Eulerian cycle is balanced because every time an ant traversing an Eulerian cycle passes through a particular vertex, it enters on one edge of the cycle and exits on the next edge. This pairs up all the edges touching each vertex, showing that half the edges on the vertex lead into it and half lead out from it.. It is a bit harder to see that every connected balanced graph contains an Eulerian cycle. To prove this fact, Euler sent an ant to randomly explore the graph under a single constraint: the ant cannot traverse a previously traversed edge. Sooner or later, the ant must get stuck at a certain node (with all outgoing edges previously traversed), and Euler noticed that because the graph is balanced, this “no exit” node is exactly the vertex where the ant started, no matter how the ant traveled through the graph. This implies that the ant has completed a cycle; if this cycle happens to traverse all edges, then the ant has found an Eulerian cycle! Otherwise, Euler sent another ant to randomly traverse unexplored edges and thereby to trace a second cycle in the graph. Euler further showed that the two cycles discovered by the two ants can be combined into a single cycle. If this (larger) cycle contains all the edges in the graph, then the two ants have together found an Eulerian cycle! If not, Euler’s method recruits a third (fourth, fifth, etc.) ant, and eventually finds an Eulerian cycle.</p><p id="P14">On modern computers, this algorithm can efficiently find Eulerian cycles in huge graphs having billions of nodes, thus avoiding the quagmire of <em>NP</em>-Completeness. Therefore, simply recasting our original problem into a slightly different framework has converted fragment assembly into a tractable computational problem; this is a commonly used strategy in computer science.</p><p id="P15">The run time required by a computer implementation of Euler’s algorithm is roughly proportional to the number of edges in the graph <em>E</em>. In the Hamiltonian approach, the time is potentially a lot larger, due to the large number of pairwise alignments needed to construct the graph, and to the <em>NP</em>-Completeness of finding a Hamiltonian cycle. A more detailed comparison of these approaches is given in ref. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R8" rid="R8" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619098" role="button" aria-expanded="false" aria-haspopup="true">8</a>.</p><p id="P16">Unfortunately, de Bruijn graphs are not a cure-all. Throughout our exposition, we have made several simplifying assumptions, which require much work to iron out formally. Yet for every apparent complication to sequence assembly, it has proven fruitful to apply some cousin of de Bruijn graphs to transform a question involving Hamiltonian cycles into a different question regarding Eulerian cycles (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#BX2" rid="BX2" class=" boxed-text">Box 2</a>). Moreover, analogs of de Bruijn graphs have been useful in many other bioinformatics problems, including antibody sequencing<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R9" rid="R9" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619100" role="button" aria-expanded="false" aria-haspopup="true">9</a></sup>, synteny block reconstruction<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R10" rid="R10" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619111" role="button" aria-expanded="false" aria-haspopup="true">10</a></sup>, and RNA assembly<sup><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R11" rid="R11" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_628619107" role="button" aria-expanded="false" aria-haspopup="true">11</a></sup>. In each of these applications, the de Bruijn graph represents the experimental data in a manner that leads to a tractable computational problem.</p><p id="P17" class="p p-last">As new sequencing technologies emerge, the best computational strategies for assembling genomes from reads may change. The factors that influence the choice of algorithms include the quantity of data (measured by read length and coverage); quality of data (including error rates); and genome structure (such as number and size of repeated regions, and GC content). Short read sequencing technologies produce very large numbers of reads, which currently favors the use of de Bruijn graphs. De Bruijn graphs are also well suited to representing genomes with repeats, whereas overlap methods need to mask repeats that are longer than the read length. However, if a future sequencing technology produces high quality reads with tens of thousands of bases, a smaller number of reads would be needed, and the pendulum could swing back towards favoring overlap-based approaches for assembly.</p></div><div id="unreferenced-ft" class="tsec sec headless whole_rhythm"><div class="unreferenced">​
                    <!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock whole_rhythm clearfix" id="F3" co-legend-rid="lgnd_F3"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F3/" target="figure" rid-figpopup="F3" rid-ob="ob-F3"><!--fig/graphic|fig/alternatives/graphic mode="anchored" m1--></a><div data-largeobj="" data-largeobj-link-rid="largeobj_idm140318149740800" class="figure"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F3/" target="figure" rid-figpopup="F3" rid-ob="ob-F3"></a><a class="inline_block ts_canvas" href="https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click%20on%20image%20to%20zoom&amp;p=PMC3&amp;id=5531759_nihms329513f3.jpg" target="tileshopwindow"><div class="ts_bar small" title="Click on image to zoom"></div><img alt="An external file that holds a picture, illustration, etc.
Object name is nihms329513f3.jpg" title="Click on image to zoom" class="tileshop" src="./Why are de Bruijn graphs useful for genome assembly__files/nihms329513f3.jpg"></a></div><div id="largeobj_idm140318149740800" class="largeobj-link align_right" style="display: none"><a target="object" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F3/?report=objectonly">Open in a separate window</a></div><div class="icnblk_cntnt" id="lgnd_F3"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/figure/F3/" target="figure" rid-figpopup="F3" rid-ob="ob-F3">Figure 3</a></div><!--caption a7--><div class="caption"><!--caption a8--><strong>De Bruijn graph</strong><p id="P21">The de Bruijn graph <em>B</em> for <em>k</em> = 4 and a 2-character alphabet composed of the digits 0 and 1. This graph has an Eulerian cycle since each node has indegree and outdegree equal to 2. Following the blue numbered edges in order 1, 2, …, 16 gives an Eulerian cycle <strong><em>0</em></strong>000, <strong><em>0</em></strong>001, <strong><em>0</em></strong>011, <strong><em>0</em></strong>110, <strong><em>1</em></strong>100, <strong><em>1</em></strong>001, <strong><em>0</em></strong>010, <strong><em>0</em></strong>101, <strong><em>1</em></strong>011, <strong><em>0</em></strong>111, <strong><em>1</em></strong>111, <strong><em>1</em></strong>110, <strong><em>1</em></strong>101, <strong><em>1</em></strong>010, <strong><em>0</em></strong>100, <strong><em>1</em></strong>000, which spells the cyclic superstring <strong><em>0000110010111101</em></strong>.</p></div></div></div></div></div><div id="SM" class="tsec sec"><a id="supplementary-material-sec"></a><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="SMtitle">Supplementary Material</h2><!--/article/body/sec/--><div class="sec suppmat" id="SD1"><h4>1</h4><div class="sup-box half_rhythm" id="d36e433"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/bin/NIHMS329513-supplement-1.pdf">Click here to view.</a><sup>(114K, pdf)</sup></div></div></div><div id="S4" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="S4title">Acknowledgments</h2><div class="sec"><p id="P18">This work was supported by grants from Howard Hughes Medical Institute (HHMI grant 52005726), the National Institutes of Health (NIH grant 3P41RR024851-02S1), and the National Science Foundation (NSF grant DMS-0718810).</p></div></div><div id="idm140318146057888aff-info" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="idm140318146057888aff-infotitle">Contributor Information</h2><p><span class="fm-affl">Phillip E. C. Compeau, </span><span class="fm-affl" lang="en" id="A1"> Department of Mathematics, University of California San Diego, La Jolla, CA, USA.</span></p><p><span class="fm-affl">Pavel A. Pevzner, </span><span class="fm-affl" lang="en" id="A2"> Department of Computer Science and Engineering, University of California San Diego, La Jolla, CA, USA.</span></p><p><span class="fm-affl">Glenn Tesler, </span><span class="fm-affl" lang="en" id="A3"> Department of Mathematics, University of California San Diego, La Jolla, CA, USA.</span></p></div><div id="idm140318185241136" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="idm140318185241136title">References</h2><div class="ref-list-sec sec" id="reference-list"><div class="ref-cit-blk half_rhythm" id="R1">1. <span class="element-citation">Euler L. Solutio problematis ad geometriam situs pertinentis. <span><span class="ref-journal">Commentarii Academiae Scientiarum Petropolitanae. </span>1741;<span class="ref-vol">8</span>:128–140.</span></span></div><div class="ref-cit-blk half_rhythm" id="R2">2. <span class="element-citation">Skiena S.  <span class="ref-journal">The Algorithm Design Manual.</span> Berlin: Springer; 2008. </span></div><div class="ref-cit-blk half_rhythm" id="R3">3. <span class="element-citation">Lander E, et al.  Initial sequencing and analysis of the human genome. <span><span class="ref-journal">Nature. </span>2001;<span class="ref-vol">409</span>:860–921.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/11237011" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R4">4. <span class="element-citation">Venter J, et al.  The sequence of the human genome. <span><span class="ref-journal">Science. </span>2001;<span class="ref-vol">291</span>:1304–1351.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/11181995" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R5">5. <span class="element-citation">Kececioglu J, Myers E. Combinatorial algorithms for DNA sequence assembly. <span><span class="ref-journal">Algorithmica. </span>1995;<span class="ref-vol">13</span>:7–51.</span></span></div><div class="ref-cit-blk half_rhythm" id="R6">6. <span class="element-citation">Adams M, et al.  The genome sequence of drosophila melanogaster. <span><span class="ref-journal">Science. </span>2000;<span class="ref-vol">287</span>:2185–2195.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/10731132" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R7">7. <span class="element-citation">Fleischmann R, et al.  Whole-genome random sequencing and assembly of haemophilus influenzae rd. <span><span class="ref-journal">Science. </span>1995;<span class="ref-vol">269</span>:496–512.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/7542800" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R8">8. <span class="element-citation">Schatz M, Delcher A, Salzberg S. Assembly of large genomes using second-generation sequencing. <span><span class="ref-journal">Genome Res. </span>2010;<span class="ref-vol">20</span>:1165–1173.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2928494/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20508146" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R9">9. <span class="element-citation">Bandeira N, Pham V, Pevzner P, Arnott D, Lill J. Automated de novo protein sequencing of monoclonal antibodies. <span><span class="ref-journal">Nat Biotechnol. </span>2008;<span class="ref-vol">26</span>:1336–1338.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2891972/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/19060866" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R10">10. <span class="element-citation">Pham S, Pevzner P. DRIMM-synteny: decomposing genomes into evolutionary conserved segments. <span><span class="ref-journal">Bioinformatics. </span>2010;<span class="ref-vol">26</span>:2509–2516.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20736338" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R11">11. <span class="element-citation">Grabherr M, et al.  Full-length transcriptome assembly from RNA-seq data without a reference genome. <span><span class="ref-journal">Nat Biotechnol. </span>2011;<span class="ref-vol">29</span>:644–652.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3571712/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/21572440" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R12">12. <span class="element-citation">de Bruijn N. A combinatorial problem. <span><span class="ref-journal">Proc. Nederl. Akad. Wetensch. </span>1946;<span class="ref-vol">49</span>:758–764.</span></span></div><div class="ref-cit-blk half_rhythm" id="R13">13. <span class="element-citation">Drmanac R, Labat I, Brukner I, Crkvenjakov R. Sequencing of megabase plus DNA by hybridization: Theory of the method. <span><span class="ref-journal">Genomics. </span>1989;<span class="ref-vol">4</span>:114–128.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/2737674" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R14">14. <span class="element-citation">Southern, E. United Kingdom patent application gb8810400. <span>1988</span></span></div><div class="ref-cit-blk half_rhythm" id="R15">15. <span class="element-citation">Lysov Y, et al.  DNA sequencing by hybridization with oligonucleotides. <span><span class="ref-journal">Doklady Academy Nauk USSR. </span>1988;<span class="ref-vol">303</span>:1508–1511.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/3250844" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R16">16. <span class="element-citation">Pevzner P. <em>l</em>-tuple DNA sequencing: Computer analysis. <span><span class="ref-journal">Journal of Biomolecular Structure and Dynamics. </span>1989;<span class="ref-vol">7</span>:63–73.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/2684223" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R17">17. <span class="element-citation">Idury R, Waterman M. A new algorithm for DNA sequence assembly. <span><span class="ref-journal">Journal of Computational Biology. </span>1995;<span class="ref-vol">2</span>:291–306.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/7497130" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R18">18. <span class="element-citation">Pevzner P, Tang H, Waterman M. An Eulerian path approach to DNA fragment assembly. <span><span class="ref-journal">Proceedings of the National Academy of Science of the United States of America. </span>2001;<span class="ref-vol">98</span>:9748–9753.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC55524/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/11504945" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R19">19. <span class="element-citation">Pevzner P, Tang H, Tesler G. De novo repeat classification and fragment assembly. <span><span class="ref-journal">Genome Res. </span>2004;<span class="ref-vol">14</span>:1786–1796.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC515325/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/15342561" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R20">20. <span class="element-citation">Chaisson M, Pevzner P. Short read fragment assembly of bacterial genomes. <span><span class="ref-journal">Genome. Res. </span>2008;<span class="ref-vol">18</span>:324–330.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2203630/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18083777" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R21">21. <span class="element-citation">Zerbino D, Birney E. Velvet: algorithms for de novo short read assembly using de Bruijn graphs. <span><span class="ref-journal">Genome. Res. </span>2008;<span class="ref-vol">18</span>:821–829.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2336801/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18349386" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R22">22. <span class="element-citation">Butler J, et al.  ALLPATHS: de novo assembly of whole-genome shotgun microreads. <span><span class="ref-journal">Genome Res. </span>2008;<span class="ref-vol">18</span>:810–820.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2336810/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18340039" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R23">23. <span class="element-citation">Simpson J, et al.  ABySS: a parallel assembler for short read sequence data. <span><span class="ref-journal">Genome. Res. </span>2009;<span class="ref-vol">19</span>:1117–1123.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2694472/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/19251739" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R24">24. <span class="element-citation">Li R, et al.  De novo assembly of human genomes with massively parallel short read sequencing. <span><span class="ref-journal">Genome Res. </span>2010;<span class="ref-vol">20</span>:265–272.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2813482/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20019144" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R25">25. <span class="element-citation">Paszkiewicz K, Studholme D. De novo assembly of short sequence reads. <span><span class="ref-journal">Brief Bioinform. </span>2010;<span class="ref-vol">11</span>:457–472.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20724458" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="R26">26. <span class="element-citation">Miller J, Koren S, Sutton G. Assembly algorithms for next-generation sequencing data. <span><span class="ref-journal">Genomics. </span>2010;<span class="ref-vol">95</span>:315–327.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2874646/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20211242" target="pmc_ext" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div></div></div><div style="display: none; width: 200px; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true" class="ui-helper-reset ui-ncbipopper-wrapper ui-ncbilinksmenu"><ul id="ui-ncbiinpagenav-2"><li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#S1title">Assembling billions of short sequencing reads into a contiguous genome is a formidable challenge</a></li><li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#S2title">Computational issues arise from alignment-based assembly</a></li><li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#S3title">Scalable assembly with de Bruijn graphs</a></li><li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#SMtitle">Supplementary Material</a></li><li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#S4title">Acknowledgments</a></li><li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#idm140318146057888aff-infotitle">Contributor Information</a></li><li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#idm140318185241136title">References</a></li></ul></div></div><!--post-content--></div>
            
            
        
            
        </div>
        <!-- Book content -->
    </div>
    
    <div id="rightcolumn" class="four_col col last">
        <!-- Custom content above discovery portlets -->
        <div class="col6">
            
        </div>
        
        <div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"><div class="format-menu"><h2>Formats:</h2><ul><li class="selected">Article</li> | <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/?report=reader">PubReader</a></li> | <li class="epub-link"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/epub/">ePub (beta)</a></li> | <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/pdf/nihms329513.pdf">PDF (1.1M)</a></li> | <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" data-citationid="PMC5531759" class="citationexporter ctxp" role="button" aria-expanded="false" aria-haspopup="true">Citation</a></li></ul></div></div><div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="share-buttons"><h2>Share</h2><ul><li class="facebook"><a href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC5531759%2F" role="button" aria-expanded="false" aria-haspopup="true" target="_blank"><img src="./Why are de Bruijn graphs useful for genome assembly__files/4047626" alt="Share on Facebook">
                             Facebook
                        </a></li><li class="twitter"><a href="https://twitter.com/intent/tweet?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC5531759%2F&amp;text=Why%20are%20de%20Bruijn%20graphs%20useful%20for%20genome%20assembly%3F" role="button" aria-expanded="false" aria-haspopup="true" target="_blank"><img src="./Why are de Bruijn graphs useful for genome assembly__files/4047627" alt="Share on Twitter">
                             Twitter
                        </a></li><li class="gplus"><a href="https://plus.google.com/share?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC5531759%2F" role="button" aria-expanded="false" aria-haspopup="true" target="_blank"><img src="./Why are de Bruijn graphs useful for genome assembly__files/4047628" alt="Share on Google Plus">
                             Google+
                        </a></li></ul></div>
        
        <div id="ajax-portlets" data-pmid="22068540" data-aiid="5531759" data-aid="5531759" data-iid="295986" data-domainid="319" data-domain="nihpa" data-accid="PMC5531759" data-md5="4a698f15d31f940a040ce2df53289015">
<div class="portlet pubmed_favoritesad">
<div class="portlet_head">
<div class="portlet_title"><h3><span>Save items</span></h3></div>
<a name="Shutter" sid="1" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" class="portlet_shutter ui-ncbitoggler-open" title="Show/hide content" remembercollapsed="true" pgsec_name="pmfavad" toggles="ui-portlet_content-4" role="button" aria-expanded="true" style="position: absolute; padding: 0px;"></a>
</div>
<div class="ui-helper-reset" aria-live="assertive"><div class="portlet_content ui-ncbitoggler ui-ncbitoggler-slave-open" id="ui-portlet_content-4" aria-hidden="false"><div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
<input type="hidden" name="absid" id="absid" value="5531759"><div id="pubmed_favoritesad" class="empty" data-db="pmc">
<span role="menubar"><div class="ui-ncbisetswitch ltd-hover" role="menuitem" aria-expanded="false" aria-haspopup="true"><a href="https://www.ncbi.nlm.nih.gov/myncbi/collections/" class="jig-ncbisetswitch blind collink ltd-hover ui-ncbisetswitch-hasStar" id="favList" data-jigconfig="destSelector:&#39;#favUL&#39;,starExists:true,createCollectionUrl:&#39;#&#39;,manageCollectionsUrl:&#39;/myncbi/collections/&#39;"><span class="star"></span>Add to Favorites</a><a class="ui-ncbisetswitch-button ltd-hover" title="View more options" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#"><span class="ui-icon ui-icon-triangle-1-s ui-ncbisetswitch-down">View more options</span></a></div></span><div style="display: none; width: 200px; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true" class="ui-helper-reset ui-ncbipopper-wrapper ui-ncbilinksmenu ui-ncbisetswitch-popper"><ul id="favUL" class="ui-ncbisetswitch-options"></ul><ul class="ui-ncbisetswitch-actions"><li class="ui-ncbisetswitch-actions-create" style="display: block;"><a class="ui-ncbisetswitch-create-collection" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#">Create collection...</a></li><li class="ui-ncbisetswitch-actions-manage" style="display: block;"><a class="ui-ncbisetswitch-manage-collections" href="https://www.ncbi.nlm.nih.gov/myncbi/collections/">Manage collections...</a></li></ul></div>
</div>
<div class="colloading">loading</div>
<span id="submenu_AddToCollections"><form method="post"><input type="submit" class="button_apply" name="addtocollection" value="1"></form></span>
</div></div></div>
</div><div class="portlet">
<div class="portlet_head">
<div class="portlet_title"><h3><span>Similar articles in PubMed</span></h3></div>
<a name="Shutter" sid="1" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" class="portlet_shutter ui-ncbitoggler-open" title="Show/hide content" remembercollapsed="true" pgsec_name="PBooksDiscovery_RA" toggles="ui-portlet_content-5" role="button" aria-expanded="true" style="position: absolute; padding: 0px;"></a>
</div>
<div class="ui-helper-reset" aria-live="assertive"><div class="portlet_content ui-ncbitoggler ui-ncbitoggler-slave-open" id="ui-portlet_content-5" aria-hidden="false">
<ul>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/15342561" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">De novo repeat classification and fragment assembly.</a><span class="source">[Genome Res. 2004]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-6" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Pevzner PA, Tang H, Tesler G. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">Genome Res. 2004 Sep; 14(9):1786-96. </em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/21999285" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal" role="button" aria-expanded="true" aria-haspopup="true">Paired de bruijn graphs: a novel approach for incorporating mate pair information into genome assemblers.</a><span class="source">[J Comput Biol. 2011]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-7" aria-live="assertive" aria-hidden="false" style="top: 428.656px; left: 431.453px; display: block;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Medvedev P, Pham S, Chaisson M, Tesler G, Pevzner P. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">J Comput Biol. 2011 Nov; 18(11):1625-34. Epub 2011 Oct 14.</em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/22803627" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Pathset graphs: a novel approach for comprehensive utilization of paired reads in genome assembly.</a><span class="source">[J Comput Biol. 2013]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-8" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Pham SK, Antipov D, Sirotkin A, Tesler G, Pevzner PA, Alekseyev MA. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">J Comput Biol. 2013 Apr; 20(4):359-71. Epub 2012 Jul 17.</em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/12188175" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Sequencing of difficult templates containing poly(A/T) tracts: closure of sequence gaps.</a><span class="source">[Biotechniques. 2002]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-9" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Langan JE, Rowbottom L, Liloglou T, Field JK, Risk JM. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">Biotechniques. 2002 Aug; 33(2):276, 278, 280. </em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/18414585" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">A scaffold analysis tool using mate-pair information in genome sequencing.</a><span class="source">[J Biomed Biotechnol. 2008]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-10" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Kim PG, Cho HG, Park K. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">J Biomed Biotechnol. 2008; 2008:675741. </em>
</div>
</li>
</ul>
<a class="seemore" href="https://www.ncbi.nlm.nih.gov/sites/entrez?db=pubmed&amp;cmd=link&amp;linkname=pubmed_pubmed_reviews&amp;uid=22068540&amp;log%24=relatedreviews&amp;logdbfrom=pmc" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Reviews&amp;rendering-type=normal">See reviews...</a><a class="seemore" href="https://www.ncbi.nlm.nih.gov/sites/entrez?db=pubmed&amp;cmd=link&amp;linkname=pubmed_pubmed&amp;uid=22068540&amp;log%24=relatedarticles&amp;logdbfrom=pmc" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Related%20Records&amp;rendering-type=normal">See all...</a>
</div></div>
</div><div class="portlet">
<div class="portlet_head">
<div class="portlet_title"><h3><span>Cited by other articles in PMC</span></h3></div>
<a name="Shutter" sid="1" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" class="portlet_shutter ui-ncbitoggler-open" title="Show/hide content" remembercollapsed="true" pgsec_name="PMCVCitedByPmcArticlesP" toggles="ui-portlet_content-11" role="button" aria-expanded="true" style="position: absolute; padding: 0px;"></a>
</div>
<div class="ui-helper-reset" aria-live="assertive"><div class="portlet_content ui-ncbitoggler ui-ncbitoggler-slave-open" id="ui-portlet_content-11" aria-hidden="false">
<ul>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5872255/" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedByInPmc&amp;TO=PMC|Article|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Succinct colored de Bruijn graphs</a><span class="source">[Bioinformatics. 2017]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-12" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Muggli MD, Bowe A, Noyes NR, Morley PS, Belk KE, Raymond R, Gagie T, Puglisi SJ, Boucher C. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">Bioinformatics. 2017 Feb 14; 33(20): 3181-3187</em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5870571/" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedByInPmc&amp;TO=PMC|Article|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">deBGR: an efficient and near-exact representation of the weighted de Bruijn graph</a><span class="source">[Bioinformatics. 2017]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-13" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Pandey P, Bender MA, Johnson R, Patro R. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">Bioinformatics. 2017 Jul 12; 33(14): i133-i141</em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5850840/" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedByInPmc&amp;TO=PMC|Article|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Phenetic Comparison of Prokaryotic Genomes Using k-mers</a><span class="source">[Molecular Biology and Evolutio...]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-14" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Déraspe M, Raymond F, Boisvert S, Culley A, Roy PH, Laviolette F, Corbeil J. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">Molecular Biology and Evolution. 2017 Jul 16; 34(10): 2716-2729</em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5826002/" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedByInPmc&amp;TO=PMC|Article|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">A Comprehensive Study of De Novo Genome Assemblers: Current Challenges and Future Prospective</a><span class="source">[Evolutionary Bioinformatics On...]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-15" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Khan AR, Pervez MT, Babar ME, Naveed N, Shoaib M. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">Evolutionary Bioinformatics Online. 2018 Feb 20; 14: 1176934318758650</em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5824816/" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedByInPmc&amp;TO=PMC|Article|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Multiple hybrid de novo genome assembly of finger millet, an orphan allotetraploid crop</a><span class="source">[DNA Research: An International...]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-16" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Hatakeyama M, Aluri S, Balachadran MT, Sivarajan SR, Patrignani A, Grüter S, Poveda L, Shimizu-Inatsugi R, Baeten J, Francoijs KJ, Nataraja KN, Reddy YA, Phadnis S, Ravikumar RL, Schlapbach R, Sreeman SM, Shimizu KK. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">DNA Research: An International Journal for Rapid Publication of Reports on Genes and Genomes. 2017 Sep 5; 25(1): 39-47</em>
</div>
</li>
</ul>
<a class="seemore" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/citedby/" ref="reftype=relart&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedByInPmc&amp;TO=PMC|Article|CitedBy&amp;rendering-type=normal">See all...</a>
</div></div>
</div><div class="portlet">
<div class="portlet_head">
<div class="portlet_title"><h3><span>Links</span></h3></div>
<a name="Shutter" sid="1" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" class="portlet_shutter ui-ncbitoggler-open" title="Show/hide content" remembercollapsed="true" pgsec_name="PMCVCitedByPmcArticlesP" toggles="ui-portlet_content-17" role="button" aria-expanded="true" style="position: absolute; padding: 0px;"></a>
</div>
<div class="ui-helper-reset" aria-live="assertive"><div class="portlet_content ui-ncbitoggler ui-ncbitoggler-slave-open" id="ui-portlet_content-17" aria-hidden="false"><ul>
<li class="brieflinkpopper">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/22068540/" ref="reftype=PubMed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|DiscoveryLinks&amp;TO=Entrez|PubMed|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">PubMed</a><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-18" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">PubMed<div class="brieflinkpopdesc">PubMed citations for these articles</div>
</div>
</li>
<li class="brieflinkpopper">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pmc/?Db=taxonomy&amp;DbFrom=pmc&amp;Cmd=Link&amp;LinkName=pmc_taxonomy&amp;IdsFromResult=5531759" ref="reftype=Taxonomy&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|DiscoveryLinks&amp;TO=Entrez|Crosslink|Taxonomy&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Taxonomy</a><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-19" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">Taxonomy<div class="brieflinkpopdesc">Related taxonomy entry</div>
</div>
</li>
</ul></div></div>
</div><div class="portlet">
<div class="portlet_head">
<div class="portlet_title"><h3><span>Recent Activity</span></h3></div>
<a name="Shutter" sid="1" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" class="portlet_shutter ui-ncbitoggler-open" title="Show/hide content" remembercollapsed="true" pgsec_name="recent_activity" toggles="ui-portlet_content-20" role="button" aria-expanded="true" style="position: absolute; padding: 0px;"></a>
</div>
<div class="ui-helper-reset" aria-live="assertive"><div class="portlet_content ui-ncbitoggler ui-ncbitoggler-slave-open" id="ui-portlet_content-20" aria-hidden="false"><div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="HTDisplay" class="">
<div class="action">
<a href="javascript:historyDisplayState(&#39;ClearHT&#39;)">Clear</a><a href="javascript:historyDisplayState(&#39;HTOff&#39;)" class="HTOn">Turn Off</a><a href="javascript:historyDisplayState(&#39;HTOn&#39;)" class="HTOff">Turn On</a>
</div>
<ul id="activity">
<li class="ra_rcd ralinkpopper two_line">
<a class="htb ralinkpopperctrl" ref="log$=activity&amp;linkpos=1" href="https://www.ncbi.nlm.nih.gov/portal/utils/pageresolver.fcgi?recordid=5b0db0939e37c8147a1a8b61">Why are de Bruijn graphs useful for genome assembly?</a><div class="ralinkpop offscreen_noflow">Why are de Bruijn graphs useful for genome assembly?<div class="brieflinkpopdesc">NIHPA Author Manuscripts. 2011 Nov 8; 29(11)987</div>
</div>
<div class="tertiary"></div>
</li>
<li class="ra_rcd ralinkpopper two_line">
<a class="htb ralinkpopperctrl" ref="log$=activity&amp;linkpos=2" href="https://www.ncbi.nlm.nih.gov/portal/utils/pageresolver.fcgi?recordid=5b0db0862fa3283f20275da7">How to apply de Bruijn graphs to genome assembly.</a><div class="ralinkpop offscreen_noflow">How to apply de Bruijn graphs to genome assembly.<div class="brieflinkpopdesc">Nat Biotechnol. 2011 Nov 8;29(11):987-91. doi: 10.1038/nbt.2023.</div>
</div>
<div class="tertiary">PubMed</div>
</li>
<li class="ra_rcd ralinkpopper two_line">
<a class="htb ralinkpopperctrl" ref="log$=activity&amp;linkpos=3" href="https://www.ncbi.nlm.nih.gov/portal/utils/pageresolver.fcgi?recordid=5b0b0a49247f773414710325">SSAHA: A Fast Search Method for Large DNA Databases</a><div class="ralinkpop offscreen_noflow">SSAHA: A Fast Search Method for Large DNA Databases<div class="brieflinkpopdesc">Genome Research. 2001 Oct; 11(10)1725</div>
</div>
<div class="tertiary"></div>
</li>
<li class="ra_rcd ralinkpopper two_line">
<a class="htb ralinkpopperctrl" ref="log$=activity&amp;linkpos=4" href="https://www.ncbi.nlm.nih.gov/portal/utils/pageresolver.fcgi?recordid=5aecce8f7f433535ee4e96cb">Illusory Recovery: Are Recovered Children With Early Language Delay at Continuin...</a><div class="ralinkpop offscreen_noflow">Illusory Recovery: Are Recovered Children With Early Language Delay at Continuing Elevated Risk?<div class="brieflinkpopdesc">UKPMC Funders Author Manuscripts. 2014 Aug; 23(3)437</div>
</div>
<div class="tertiary"></div>
</li>
</ul>
<p class="HTOn">Your browsing activity is empty.</p>
<p class="HTOff">Activity recording is turned off.</p>
<p id="turnOn" class="HTOff"><a href="javascript:historyDisplayState(&#39;HTOn&#39;)">Turn recording back on</a></p>
<a class="seemore" href="https://www.ncbi.nlm.nih.gov/sites/myncbi/recentactivity">See more...</a>
</div></div></div>
</div>
<div>
<div class="portlet brieflink pmc_para_cit" id="crb--P3" name="crb--P3" rid="P3" style="position: absolute; visibility: visible; width: 370px; top: 1334px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-21"><ul>
<li class="two_line" reference_id="R3">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/11237011/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Initial sequencing and analysis of the human genome.</a><span class="one_line_source">[Nature.  2001]</span><div class="alt-note">
<div class="authors">Lander ES, Linton LM, Birren B, Nusbaum C, Zody MC, Baldwin J, Devon K, Dewar K, Doyle M, FitzHugh W, Funke R, Gage D, Harris K, Heaford A, Howland J, Kann L, Lehoczky J, LeVine R, McEwan P, McKernan K, Meldrim J, Mesirov JP, Miranda C, Morris W, Naylor J, Raymond C, Rosetti M, Santos R, Sheridan A, Sougnez C, Stange-Thomann Y, Stojanovic N, Subramanian A, Wyman D, Rogers J, Sulston J, Ainscough R, Beck S, Bentley D, Burton J, Clee C, Carter N, Coulson A, Deadman R, Deloukas P, Dunham A, Dunham I, Durbin R, French L, Grafham D, Gregory S, Hubbard T, Humphray S, Hunt A, Jones M, Lloyd C, McMurray A, Matthews L, Mercer S, Milne S, Mullikin JC, Mungall A, Plumb R, Ross M, Shownkeen R, Sims S, Waterston RH, Wilson RK, Hillier LW, McPherson JD, Marra MA, Mardis ER, Fulton LA, Chinwalla AT, Pepin KH, Gish WR, Chissoe SL, Wendl MC, Delehaunty KD, Miner TL, Delehaunty A, Kramer JB, Cook LL, Fulton RS, Johnson DL, Minx PJ, Clifton SW, Hawkins T, Branscomb E, Predki P, Richardson P, Wenning S, Slezak T, Doggett N, Cheng JF, Olsen A, Lucas S, Elkin C, Uberbacher E, Frazier M, Gibbs RA, Muzny DM, Scherer SE, Bouck JB, Sodergren EJ, Worley KC, Rives CM, Gorrell JH, Metzker ML, Naylor SL, Kucherlapati RS, Nelson DL, Weinstock GM, Sakaki Y, Fujiyama A, Hattori M, Yada T, Toyoda A, Itoh T, Kawagoe C, Watanabe H, Totoki Y, Taylor T, Weissenbach J, Heilig R, Saurin W, Artiguenave F, Brottier P, Bruls T, Pelletier E, Robert C, Wincker P, Smith DR, Doucette-Stamm L, Rubenfield M, Weinstock K, Lee HM, Dubois J, Rosenthal A, Platzer M, Nyakatura G, Taudien S, Rump A, Yang H, Yu J, Wang J, Huang G, Gu J, Hood L, Rowen L, Madan A, Qin S, Davis RW, Federspiel NA, Abola AP, Proctor MJ, Myers RM, Schmutz J, Dickson M, Grimwood J, Cox DR, Olson MV, Kaul R, Raymond C, Shimizu N, Kawasaki K, Minoshima S, Evans GA, Athanasiou M, Schultz R, Roe BA, Chen F, Pan H, Ramser J, Lehrach H, Reinhardt R, McCombie WR, de la Bastide M, Dedhia N, Blöcker H, Hornischer K, Nordsiek G, Agarwala R, Aravind L, Bailey JA, Bateman A, Batzoglou S, Birney E, Bork P, Brown DG, Burge CB, Cerutti L, Chen HC, Church D, Clamp M, Copley RR, Doerks T, Eddy SR, Eichler EE, Furey TS, Galagan J, Gilbert JG, Harmon C, Hayashizaki Y, Haussler D, Hermjakob H, Hokamp K, Jang W, Johnson LS, Jones TA, Kasif S, Kaspryzk A, Kennedy S, Kent WJ, Kitts P, Koonin EV, Korf I, Kulp D, Lancet D, Lowe TM, McLysaght A, Mikkelsen T, Moran JV, Mulder N, Pollara VJ, Ponting CP, Schuler G, Schultz J, Slater G, Smit AF, Stupka E, Szustakowki J, Thierry-Mieg D, Thierry-Mieg J, Wagner L, Wallis J, Wheeler R, Williams A, Wolf YI, Wolfe KH, Yang SP, Yeh RF, Collins F, Guyer MS, Peterson J, Felsenfeld A, Wetterstrand KA, Patrinos A, Morgan MJ, de Jong P, Catanese JJ, Osoegawa K, Shizuya H, Choi S, Chen YJ, Szustakowki J, International Human Genome Sequencing Consortium.</div>
<div class="citation">Nature. 2001 Feb 15; 409(6822):860-921.</div>
</div>
</li>
<li class="two_line" reference_id="R4">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/11181995/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">The sequence of the human genome.</a><span class="one_line_source">[Science.  2001]</span><div class="alt-note">
<div class="authors">Venter JC, Adams MD, Myers EW, Li PW, Mural RJ, Sutton GG, Smith HO, Yandell M, Evans CA, Holt RA, Gocayne JD, Amanatides P, Ballew RM, Huson DH, Wortman JR, Zhang Q, Kodira CD, Zheng XH, Chen L, Skupski M, Subramanian G, Thomas PD, Zhang J, Gabor Miklos GL, Nelson C, Broder S, Clark AG, Nadeau J, McKusick VA, Zinder N, Levine AJ, Roberts RJ, Simon M, Slayman C, Hunkapiller M, Bolanos R, Delcher A, Dew I, Fasulo D, Flanigan M, Florea L, Halpern A, Hannenhalli S, Kravitz S, Levy S, Mobarry C, Reinert K, Remington K, Abu-Threideh J, Beasley E, Biddick K, Bonazzi V, Brandon R, Cargill M, Chandramouliswaran I, Charlab R, Chaturvedi K, Deng Z, Di Francesco V, Dunn P, Eilbeck K, Evangelista C, Gabrielian AE, Gan W, Ge W, Gong F, Gu Z, Guan P, Heiman TJ, Higgins ME, Ji RR, Ke Z, Ketchum KA, Lai Z, Lei Y, Li Z, Li J, Liang Y, Lin X, Lu F, Merkulov GV, Milshina N, Moore HM, Naik AK, Narayan VA, Neelam B, Nusskern D, Rusch DB, Salzberg S, Shao W, Shue B, Sun J, Wang Z, Wang A, Wang X, Wang J, Wei M, Wides R, Xiao C, Yan C, Yao A, Ye J, Zhan M, Zhang W, Zhang H, Zhao Q, Zheng L, Zhong F, Zhong W, Zhu S, Zhao S, Gilbert D, Baumhueter S, Spier G, Carter C, Cravchik A, Woodage T, Ali F, An H, Awe A, Baldwin D, Baden H, Barnstead M, Barrow I, Beeson K, Busam D, Carver A, Center A, Cheng ML, Curry L, Danaher S, Davenport L, Desilets R, Dietz S, Dodson K, Doup L, Ferriera S, Garg N, Gluecksmann A, Hart B, Haynes J, Haynes C, Heiner C, Hladun S, Hostin D, Houck J, Howland T, Ibegwam C, Johnson J, Kalush F, Kline L, Koduru S, Love A, Mann F, May D, McCawley S, McIntosh T, McMullen I, Moy M, Moy L, Murphy B, Nelson K, Pfannkoch C, Pratts E, Puri V, Qureshi H, Reardon M, Rodriguez R, Rogers YH, Romblad D, Ruhfel B, Scott R, Sitter C, Smallwood M, Stewart E, Strong R, Suh E, Thomas R, Tint NN, Tse S, Vech C, Wang G, Wetter J, Williams S, Williams M, Windsor S, Winn-Deen E, Wolfe K, Zaveri J, Zaveri K, Abril JF, Guigó R, Campbell MJ, Sjolander KV, Karlak B, Kejariwal A, Mi H, Lazareva B, Hatton T, Narechania A, Diemer K, Muruganujan A, Guo N, Sato S, Bafna V, Istrail S, Lippert R, Schwartz R, Walenz B, Yooseph S, Allen D, Basu A, Baxendale J, Blick L, Caminha M, Carnes-Stine J, Caulk P, Chiang YH, Coyne M, Dahlke C, Mays A, Dombroski M, Donnelly M, Ely D, Esparham S, Fosler C, Gire H, Glanowski S, Glasser K, Glodek A, Gorokhov M, Graham K, Gropman B, Harris M, Heil J, Henderson S, Hoover J, Jennings D, Jordan C, Jordan J, Kasha J, Kagan L, Kraft C, Levitsky A, Lewis M, Liu X, Lopez J, Ma D, Majoros W, McDaniel J, Murphy S, Newman M, Nguyen T, Nguyen N, Nodell M, Pan S, Peck J, Peterson M, Rowe W, Sanders R, Scott J, Simpson M, Smith T, Sprague A, Stockwell T, Turner R, Venter E, Wang M, Wen M, Wu D, Wu M, Xia A, Zandieh A, Zhu X</div>
<div class="citation">Science. 2001 Feb 16; 291(5507):1304-51.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--P7" name="crb--P7" rid="P7" style="position: absolute; visibility: visible; width: 370px; top: 4626px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-22"><ul>
<li class="two_line" reference_id="R6">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/10731132/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">The genome sequence of Drosophila melanogaster.</a><span class="one_line_source">[Science.  2000]</span><div class="alt-note">
<div class="authors">Adams MD, Celniker SE, Holt RA, Evans CA, Gocayne JD, Amanatides PG, Scherer SE, Li PW, Hoskins RA, Galle RF, George RA, Lewis SE, Richards S, Ashburner M, Henderson SN, Sutton GG, Wortman JR, Yandell MD, Zhang Q, Chen LX, Brandon RC, Rogers YH, Blazej RG, Champe M, Pfeiffer BD, Wan KH, Doyle C, Baxter EG, Helt G, Nelson CR, Gabor GL, Abril JF, Agbayani A, An HJ, Andrews-Pfannkoch C, Baldwin D, Ballew RM, Basu A, Baxendale J, Bayraktaroglu L, Beasley EM, Beeson KY, Benos PV, Berman BP, Bhandari D, Bolshakov S, Borkova D, Botchan MR, Bouck J, Brokstein P, Brottier P, Burtis KC, Busam DA, Butler H, Cadieu E, Center A, Chandra I, Cherry JM, Cawley S, Dahlke C, Davenport LB, Davies P, de Pablos B, Delcher A, Deng Z, Mays AD, Dew I, Dietz SM, Dodson K, Doup LE, Downes M, Dugan-Rocha S, Dunkov BC, Dunn P, Durbin KJ, Evangelista CC, Ferraz C, Ferriera S, Fleischmann W, Fosler C, Gabrielian AE, Garg NS, Gelbart WM, Glasser K, Glodek A, Gong F, Gorrell JH, Gu Z, Guan P, Harris M, Harris NL, Harvey D, Heiman TJ, Hernandez JR, Houck J, Hostin D, Houston KA, Howland TJ, Wei MH, Ibegwam C, Jalali M, Kalush F, Karpen GH, Ke Z, Kennison JA, Ketchum KA, Kimmel BE, Kodira CD, Kraft C, Kravitz S, Kulp D, Lai Z, Lasko P, Lei Y, Levitsky AA, Li J, Li Z, Liang Y, Lin X, Liu X, Mattei B, McIntosh TC, McLeod MP, McPherson D, Merkulov G, Milshina NV, Mobarry C, Morris J, Moshrefi A, Mount SM, Moy M, Murphy B, Murphy L, Muzny DM, Nelson DL, Nelson DR, Nelson KA, Nixon K, Nusskern DR, Pacleb JM, Palazzolo M, Pittman GS, Pan S, Pollard J, Puri V, Reese MG, Reinert K, Remington K, Saunders RD, Scheeler F, Shen H, Shue BC, Sidén-Kiamos I, Simpson M, Skupski MP, Smith T, Spier E, Spradling AC, Stapleton M, Strong R, Sun E, Svirskas R, Tector C, Turner R, Venter E, Wang AH, Wang X, Wang ZY, Wassarman DA, Weinstock GM, Weissenbach J, Williams SM, WoodageT, Worley KC, Wu D, Yang S, Yao QA, Ye J, Yeh RF, Zaveri JS, Zhan M, Zhang G, Zhao Q, Zheng L, Zheng XH, Zhong FN, Zhong W, Zhou X, Zhu S, Zhu X, Smith HO, Gibbs RA, Myers EW, Rubin GM, Venter JC</div>
<div class="citation">Science. 2000 Mar 24; 287(5461):2185-95.</div>
</div>
</li>
<li class="two_line" reference_id="R7">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/7542800/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Whole-genome random sequencing and assembly of Haemophilus influenzae Rd.</a><span class="one_line_source">[Science.  1995]</span><div class="alt-note">
<div class="authors">Fleischmann RD, Adams MD, White O, Clayton RA, Kirkness EF, Kerlavage AR, Bult CJ, Tomb JF, Dougherty BA, Merrick JM</div>
<div class="citation">Science. 1995 Jul 28; 269(5223):496-512.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--P15" name="crb--P15" rid="P15" style="position: absolute; visibility: visible; width: 370px; top: 6915px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-23"><ul><li class="two_line" reference_id="R8">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/20508146/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Assembly of large genomes using second-generation sequencing.</a><span class="one_line_source">[Genome Res.  2010]</span><div class="alt-note">
<div class="authors">Schatz MC, Delcher AL, Salzberg SL</div>
<div class="citation">Genome Res. 2010 Sep; 20(9):1165-73.</div>
</div>
</li></ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--P16" name="crb--P16" rid="P16" style="position: absolute; visibility: visible; width: 370px; top: 7010px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-24"><ul>
<li class="two_line" reference_id="R9">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/19060866/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Automated de novo protein sequencing of monoclonal antibodies.</a><span class="one_line_source">[Nat Biotechnol.  2008]</span><div class="alt-note">
<div class="authors">Bandeira N, Pham V, Pevzner P, Arnott D, Lill JR</div>
<div class="citation">Nat Biotechnol. 2008 Dec; 26(12):1336-8.</div>
</div>
</li>
<li class="two_line" reference_id="R10">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/20736338/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">DRIMM-Synteny: decomposing genomes into evolutionary conserved segments.</a><span class="one_line_source">[Bioinformatics.  2010]</span><div class="alt-note">
<div class="authors">Pham SK, Pevzner PA</div>
<div class="citation">Bioinformatics. 2010 Oct 15; 26(20):2509-16.</div>
</div>
</li>
<li class="two_line" reference_id="R11">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/21572440/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Full-length transcriptome assembly from RNA-Seq data without a reference genome.</a><span class="one_line_source">[Nat Biotechnol.  2011]</span><div class="alt-note">
<div class="authors">Grabherr MG, Haas BJ, Yassour M, Levin JZ, Thompson DA, Amit I, Adiconis X, Fan L, Raychowdhury R, Zeng Q, Chen Z, Mauceli E, Hacohen N, Gnirke A, Rhind N, di Palma F, Birren BW, Nusbaum C, Lindblad-Toh K, Friedman N, Regev A</div>
<div class="citation">Nat Biotechnol. 2011 May 15; 29(7):644-52.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--P24" name="crb--P24" rid="P24" style="position: absolute; visibility: visible; width: 370px; top: 5781px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-25"><ul>
<li class="two_line" reference_id="R13">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/2737674/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Sequencing of megabase plus DNA by hybridization: theory of the method.</a><span class="one_line_source">[Genomics.  1989]</span><div class="alt-note">
<div class="authors">Drmanac R, Labat I, Brukner I, Crkvenjakov R</div>
<div class="citation">Genomics. 1989 Feb; 4(2):114-28.</div>
</div>
</li>
<li class="two_line" reference_id="R15">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/3250844/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">[Determination of the nucleotide sequence of DNA using hybridization with oligonucleotides. A new method].</a><span class="one_line_source">[Dokl Akad Nauk SSSR.  1988]</span><div class="alt-note">
<div class="authors">Lysov IuP, Florent'ev VL, Khorlin AA, Khrapko KR, Shik VV</div>
<div class="citation">Dokl Akad Nauk SSSR. 1988; 303(6):1508-11.</div>
</div>
</li>
<li class="two_line" reference_id="R16">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/2684223/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal"><span class="invert">Review</span> 1-Tuple DNA sequencing: computer analysis.</a><span class="one_line_source">[J Biomol Struct Dyn.  1989]</span><div class="alt-note">
<div class="authors">Pevzner PA</div>
<div class="citation">J Biomol Struct Dyn. 1989 Aug; 7(1):63-73.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--P28" name="crb--P28" rid="P28" style="position: absolute; visibility: visible; width: 370px; top: 3090px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-26"><ul><li class="two_line" reference_id="R17">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/7497130/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">A new algorithm for DNA sequence assembly.</a><span class="one_line_source">[J Comput Biol.  1995]</span><div class="alt-note">
<div class="authors">Idury RM, Waterman MS</div>
<div class="citation">J Comput Biol. 1995 Summer; 2(2):291-306.</div>
</div>
</li></ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--P29" name="crb--P29" rid="P29" style="position: absolute; visibility: visible; width: 370px; top: 3322px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-27"><ul>
<li class="two_line" reference_id="R18">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/11504945/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">An Eulerian path approach to DNA fragment assembly.</a><span class="one_line_source">[Proc Natl Acad Sci U S A.  2001]</span><div class="alt-note">
<div class="authors">Pevzner PA, Tang H, Waterman MS</div>
<div class="citation">Proc Natl Acad Sci U S A. 2001 Aug 14; 98(17):9748-53.</div>
</div>
</li>
<li class="two_line" reference_id="R19">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/15342561/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">De novo repeat classification and fragment assembly.</a><span class="one_line_source">[Genome Res.  2004]</span><div class="alt-note">
<div class="authors">Pevzner PA, Tang H, Tesler G</div>
<div class="citation">Genome Res. 2004 Sep; 14(9):1786-96.</div>
</div>
</li>
<li class="two_line" reference_id="R20">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18083777/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Short read fragment assembly of bacterial genomes.</a><span class="one_line_source">[Genome Res.  2008]</span><div class="alt-note">
<div class="authors">Chaisson MJ, Pevzner PA</div>
<div class="citation">Genome Res. 2008 Feb; 18(2):324-30.</div>
</div>
</li>
<li class="two_line" reference_id="R21">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18349386/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Velvet: algorithms for de novo short read assembly using de Bruijn graphs.</a><span class="one_line_source">[Genome Res.  2008]</span><div class="alt-note">
<div class="authors">Zerbino DR, Birney E</div>
<div class="citation">Genome Res. 2008 May; 18(5):821-9.</div>
</div>
</li>
<li class="two_line" reference_id="R22">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18340039/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">ALLPATHS: de novo assembly of whole-genome shotgun microreads.</a><span class="one_line_source">[Genome Res.  2008]</span><div class="alt-note">
<div class="authors">Butler J, MacCallum I, Kleber M, Shlyakhter IA, Belmonte MK, Lander ES, Nusbaum C, Jaffe DB</div>
<div class="citation">Genome Res. 2008 May; 18(5):810-20.</div>
</div>
</li>
<li class="two_line" reference_id="R23" style="display: none;">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/19251739/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">ABySS: a parallel assembler for short read sequence data.</a><span class="one_line_source">[Genome Res.  2009]</span><div class="alt-note">
<div class="authors">Simpson JT, Wong K, Jackman SD, Schein JE, Jones SJ, Birol I</div>
<div class="citation">Genome Res. 2009 Jun; 19(6):1117-23.</div>
</div>
</li>
<li class="two_line" reference_id="R24" style="display: none;">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/20019144/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">De novo assembly of human genomes with massively parallel short read sequencing.</a><span class="one_line_source">[Genome Res.  2010]</span><div class="alt-note">
<div class="authors">Li R, Zhu H, Ruan J, Qian W, Fang X, Shi Z, Li Y, Li S, Shan G, Kristiansen K, Li S, Yang H, Wang J, Wang J</div>
<div class="citation">Genome Res. 2010 Feb; 20(2):265-72.</div>
</div>
</li>
<li class="two_line" reference_id="R25" style="display: none;">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/20724458/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal"><span class="invert">Review</span> De novo assembly of short sequence reads.</a><span class="one_line_source">[Brief Bioinform.  2010]</span><div class="alt-note">
<div class="authors">Paszkiewicz K, Studholme DJ</div>
<div class="citation">Brief Bioinform. 2010 Sep; 11(5):457-72.</div>
</div>
</li>
<li class="two_line" reference_id="R26" style="display: none;">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/20211242/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal"><span class="invert">Review</span> Assembly algorithms for next-generation sequencing data.</a><span class="one_line_source">[Genomics.  2010]</span><div class="alt-note">
<div class="authors">Miller JR, Koren S, Sutton G</div>
<div class="citation">Genomics. 2010 Jun; 95(6):315-27.</div>
</div>
</li>
</ul><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/" class="seemore">See more ...</a></div>
</div>
</div>   
    </div>
                
        <!-- Custom content below discovery portlets -->
        <div class="col7">
            
        </div>
    </div>
</div>

<!-- Custom content after all -->
<div class="col8">
    
</div>
<div class="col9">
    
</div>

<script src="./Why are de Bruijn graphs useful for genome assembly__files/jquery.scrollTo-1.4.2.js"></script>
<script>
    (function($){
        $('.skiplink').each(function(i, item){
            var href = $($(item).attr('href'));
            href.attr('tabindex', '-1').addClass('skiptarget'); // ensure the target can receive focus
            $(item).on('click', function(event){
                event.preventDefault();
                $.scrollTo(href, 0, {
                    onAfter: function(){
                        href.focus();
                    }
                });
            });
        });
    })(jQuery);
</script>



<div id="body-link-poppers"><div id="body-link-popper-R1" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">1. <span class="element-citation">Euler L. Solutio problematis ad geometriam situs pertinentis. <span><span class="ref-journal">Commentarii Academiae Scientiarum Petropolitanae. </span>1741;<span class="ref-vol">8</span>:128–140.</span></span> [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R1">Ref list</a>]</div><div id="body-link-popper-R2" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">2. <span class="element-citation">Skiena S.  <span class="ref-journal">The Algorithm Design Manual.</span> Berlin: Springer; 2008. </span> [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R2">Ref list</a>]</div><div id="body-link-popper-R3" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Initial sequencing and analysis of the human genome.<div class="authors">Lander ES, Linton LM, Birren B, Nusbaum C, Zody MC, Baldwin J, Devon K, Dewar K, Doyle M, FitzHugh W, Funke R, Gage D, Harris K, Heaford A, Howland J, Kann L, Lehoczky J, LeVine R, McEwan P, McKernan K, Meldrim J, Mesirov JP, Miranda C, Morris W, Naylor J, Raymond C, Rosetti M, Santos R, Sheridan A, Sougnez C, Stange-Thomann Y, Stojanovic N, Subramanian A, Wyman D, Rogers J, Sulston J, Ainscough R, Beck S, Bentley D, Burton J, Clee C, Carter N, Coulson A, Deadman R, Deloukas P, Dunham A, Dunham I, Durbin R, French L, Grafham D, Gregory S, Hubbard T, Humphray S, Hunt A, Jones M, Lloyd C, McMurray A, Matthews L, Mercer S, Milne S, Mullikin JC, Mungall A, Plumb R, Ross M, Shownkeen R, Sims S, Waterston RH, Wilson RK, Hillier LW, McPherson JD, Marra MA, Mardis ER, Fulton LA, Chinwalla AT, Pepin KH, Gish WR, Chissoe SL, Wendl MC, Delehaunty KD, Miner TL, Delehaunty A, Kramer JB, Cook LL, Fulton RS, Johnson DL, Minx PJ, Clifton SW, Hawkins T, Branscomb E, Predki P, Richardson P, Wenning S, Slezak T, Doggett N, Cheng JF, Olsen A, Lucas S, Elkin C, Uberbacher E, Frazier M, Gibbs RA, Muzny DM, Scherer SE, Bouck JB, Sodergren EJ, Worley KC, Rives CM, Gorrell JH, Metzker ML, Naylor SL, Kucherlapati RS, Nelson DL, Weinstock GM, Sakaki Y, Fujiyama A, Hattori M, Yada T, Toyoda A, Itoh T, Kawagoe C, Watanabe H, Totoki Y, Taylor T, Weissenbach J, Heilig R, Saurin W, Artiguenave F, Brottier P, Bruls T, Pelletier E, Robert C, Wincker P, Smith DR, Doucette-Stamm L, Rubenfield M, Weinstock K, Lee HM, Dubois J, Rosenthal A, Platzer M, Nyakatura G, Taudien S, Rump A, Yang H, Yu J, Wang J, Huang G, Gu J, Hood L, Rowen L, Madan A, Qin S, Davis RW, Federspiel NA, Abola AP, Proctor MJ, Myers RM, Schmutz J, Dickson M, Grimwood J, Cox DR, Olson MV, Kaul R, Raymond C, Shimizu N, Kawasaki K, Minoshima S, Evans GA, Athanasiou M, Schultz R, Roe BA, Chen F, Pan H, Ramser J, Lehrach H, Reinhardt R, McCombie WR, de la Bastide M, Dedhia N, Blöcker H, Hornischer K, Nordsiek G, Agarwala R, Aravind L, Bailey JA, Bateman A, Batzoglou S, Birney E, Bork P, Brown DG, Burge CB, Cerutti L, Chen HC, Church D, Clamp M, Copley RR, Doerks T, Eddy SR, Eichler EE, Furey TS, Galagan J, Gilbert JG, Harmon C, Hayashizaki Y, Haussler D, Hermjakob H, Hokamp K, Jang W, Johnson LS, Jones TA, Kasif S, Kaspryzk A, Kennedy S, Kent WJ, Kitts P, Koonin EV, Korf I, Kulp D, Lancet D, Lowe TM, McLysaght A, Mikkelsen T, Moran JV, Mulder N, Pollara VJ, Ponting CP, Schuler G, Schultz J, Slater G, Smit AF, Stupka E, Szustakowki J, Thierry-Mieg D, Thierry-Mieg J, Wagner L, Wallis J, Wheeler R, Williams A, Wolf YI, Wolfe KH, Yang SP, Yeh RF, Collins F, Guyer MS, Peterson J, Felsenfeld A, Wetterstrand KA, Patrinos A, Morgan MJ, de Jong P, Catanese JJ, Osoegawa K, Shizuya H, Choi S, Chen YJ, Szustakowki J, International Human Genome Sequencing Consortium.</div><div class="citation">Nature. 2001 Feb 15; 409(6822):860-921.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/11237011/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R3">Ref list</a>]</p></div><div id="body-link-popper-R4" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">The sequence of the human genome.<div class="authors">Venter JC, Adams MD, Myers EW, Li PW, Mural RJ, Sutton GG, Smith HO, Yandell M, Evans CA, Holt RA, Gocayne JD, Amanatides P, Ballew RM, Huson DH, Wortman JR, Zhang Q, Kodira CD, Zheng XH, Chen L, Skupski M, Subramanian G, Thomas PD, Zhang J, Gabor Miklos GL, Nelson C, Broder S, Clark AG, Nadeau J, McKusick VA, Zinder N, Levine AJ, Roberts RJ, Simon M, Slayman C, Hunkapiller M, Bolanos R, Delcher A, Dew I, Fasulo D, Flanigan M, Florea L, Halpern A, Hannenhalli S, Kravitz S, Levy S, Mobarry C, Reinert K, Remington K, Abu-Threideh J, Beasley E, Biddick K, Bonazzi V, Brandon R, Cargill M, Chandramouliswaran I, Charlab R, Chaturvedi K, Deng Z, Di Francesco V, Dunn P, Eilbeck K, Evangelista C, Gabrielian AE, Gan W, Ge W, Gong F, Gu Z, Guan P, Heiman TJ, Higgins ME, Ji RR, Ke Z, Ketchum KA, Lai Z, Lei Y, Li Z, Li J, Liang Y, Lin X, Lu F, Merkulov GV, Milshina N, Moore HM, Naik AK, Narayan VA, Neelam B, Nusskern D, Rusch DB, Salzberg S, Shao W, Shue B, Sun J, Wang Z, Wang A, Wang X, Wang J, Wei M, Wides R, Xiao C, Yan C, Yao A, Ye J, Zhan M, Zhang W, Zhang H, Zhao Q, Zheng L, Zhong F, Zhong W, Zhu S, Zhao S, Gilbert D, Baumhueter S, Spier G, Carter C, Cravchik A, Woodage T, Ali F, An H, Awe A, Baldwin D, Baden H, Barnstead M, Barrow I, Beeson K, Busam D, Carver A, Center A, Cheng ML, Curry L, Danaher S, Davenport L, Desilets R, Dietz S, Dodson K, Doup L, Ferriera S, Garg N, Gluecksmann A, Hart B, Haynes J, Haynes C, Heiner C, Hladun S, Hostin D, Houck J, Howland T, Ibegwam C, Johnson J, Kalush F, Kline L, Koduru S, Love A, Mann F, May D, McCawley S, McIntosh T, McMullen I, Moy M, Moy L, Murphy B, Nelson K, Pfannkoch C, Pratts E, Puri V, Qureshi H, Reardon M, Rodriguez R, Rogers YH, Romblad D, Ruhfel B, Scott R, Sitter C, Smallwood M, Stewart E, Strong R, Suh E, Thomas R, Tint NN, Tse S, Vech C, Wang G, Wetter J, Williams S, Williams M, Windsor S, Winn-Deen E, Wolfe K, Zaveri J, Zaveri K, Abril JF, Guigó R, Campbell MJ, Sjolander KV, Karlak B, Kejariwal A, Mi H, Lazareva B, Hatton T, Narechania A, Diemer K, Muruganujan A, Guo N, Sato S, Bafna V, Istrail S, Lippert R, Schwartz R, Walenz B, Yooseph S, Allen D, Basu A, Baxendale J, Blick L, Caminha M, Carnes-Stine J, Caulk P, Chiang YH, Coyne M, Dahlke C, Mays A, Dombroski M, Donnelly M, Ely D, Esparham S, Fosler C, Gire H, Glanowski S, Glasser K, Glodek A, Gorokhov M, Graham K, Gropman B, Harris M, Heil J, Henderson S, Hoover J, Jennings D, Jordan C, Jordan J, Kasha J, Kagan L, Kraft C, Levitsky A, Lewis M, Liu X, Lopez J, Ma D, Majoros W, McDaniel J, Murphy S, Newman M, Nguyen T, Nguyen N, Nodell M, Pan S, Peck J, Peterson M, Rowe W, Sanders R, Scott J, Simpson M, Smith T, Sprague A, Stockwell T, Turner R, Venter E, Wang M, Wen M, Wu D, Wu M, Xia A, Zandieh A, Zhu X</div><div class="citation">Science. 2001 Feb 16; 291(5507):1304-51.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/11181995/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R4">Ref list</a>]</p></div><div id="body-link-popper-R17" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">A new algorithm for DNA sequence assembly.<div class="authors">Idury RM, Waterman MS</div><div class="citation">J Comput Biol. 1995 Summer; 2(2):291-306.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/7497130/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R17">Ref list</a>]</p></div><div id="body-link-popper-R18" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">An Eulerian path approach to DNA fragment assembly.<div class="authors">Pevzner PA, Tang H, Waterman MS</div><div class="citation">Proc Natl Acad Sci U S A. 2001 Aug 14; 98(17):9748-53.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/11504945/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R18">Ref list</a>]</p></div><div id="body-link-popper-R19" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">De novo repeat classification and fragment assembly.<div class="authors">Pevzner PA, Tang H, Tesler G</div><div class="citation">Genome Res. 2004 Sep; 14(9):1786-96.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/15342561/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R19">Ref list</a>]</p></div><div id="body-link-popper-R20" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Short read fragment assembly of bacterial genomes.<div class="authors">Chaisson MJ, Pevzner PA</div><div class="citation">Genome Res. 2008 Feb; 18(2):324-30.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18083777/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R20">Ref list</a>]</p></div><div id="body-link-popper-R21" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Velvet: algorithms for de novo short read assembly using de Bruijn graphs.<div class="authors">Zerbino DR, Birney E</div><div class="citation">Genome Res. 2008 May; 18(5):821-9.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18349386/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R21">Ref list</a>]</p></div><div id="body-link-popper-R22" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">ALLPATHS: de novo assembly of whole-genome shotgun microreads.<div class="authors">Butler J, MacCallum I, Kleber M, Shlyakhter IA, Belmonte MK, Lander ES, Nusbaum C, Jaffe DB</div><div class="citation">Genome Res. 2008 May; 18(5):810-20.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18340039/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R22">Ref list</a>]</p></div><div id="body-link-popper-R23" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">ABySS: a parallel assembler for short read sequence data.<div class="authors">Simpson JT, Wong K, Jackman SD, Schein JE, Jones SJ, Birol I</div><div class="citation">Genome Res. 2009 Jun; 19(6):1117-23.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/19251739/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R23">Ref list</a>]</p></div><div id="body-link-popper-R24" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">De novo assembly of human genomes with massively parallel short read sequencing.<div class="authors">Li R, Zhu H, Ruan J, Qian W, Fang X, Shi Z, Li Y, Li S, Shan G, Kristiansen K, Li S, Yang H, Wang J, Wang J</div><div class="citation">Genome Res. 2010 Feb; 20(2):265-72.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20019144/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R24">Ref list</a>]</p></div><div id="body-link-popper-R25" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true"><span class="invert">Review</span> De novo assembly of short sequence reads.<div class="authors">Paszkiewicz K, Studholme DJ</div><div class="citation">Brief Bioinform. 2010 Sep; 11(5):457-72.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20724458/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R25">Ref list</a>]</p></div><div id="body-link-popper-R26" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true"><span class="invert">Review</span> Assembly algorithms for next-generation sequencing data.<div class="authors">Miller JR, Koren S, Sutton G</div><div class="citation">Genomics. 2010 Jun; 95(6):315-27.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20211242/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R26">Ref list</a>]</p></div><div id="body-link-popper-R5" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">5. <span class="element-citation">Kececioglu J, Myers E. Combinatorial algorithms for DNA sequence assembly. <span><span class="ref-journal">Algorithmica. </span>1995;<span class="ref-vol">13</span>:7–51.</span></span> [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R5">Ref list</a>]</div><div id="body-link-popper-R6" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">The genome sequence of Drosophila melanogaster.<div class="authors">Adams MD, Celniker SE, Holt RA, Evans CA, Gocayne JD, Amanatides PG, Scherer SE, Li PW, Hoskins RA, Galle RF, George RA, Lewis SE, Richards S, Ashburner M, Henderson SN, Sutton GG, Wortman JR, Yandell MD, Zhang Q, Chen LX, Brandon RC, Rogers YH, Blazej RG, Champe M, Pfeiffer BD, Wan KH, Doyle C, Baxter EG, Helt G, Nelson CR, Gabor GL, Abril JF, Agbayani A, An HJ, Andrews-Pfannkoch C, Baldwin D, Ballew RM, Basu A, Baxendale J, Bayraktaroglu L, Beasley EM, Beeson KY, Benos PV, Berman BP, Bhandari D, Bolshakov S, Borkova D, Botchan MR, Bouck J, Brokstein P, Brottier P, Burtis KC, Busam DA, Butler H, Cadieu E, Center A, Chandra I, Cherry JM, Cawley S, Dahlke C, Davenport LB, Davies P, de Pablos B, Delcher A, Deng Z, Mays AD, Dew I, Dietz SM, Dodson K, Doup LE, Downes M, Dugan-Rocha S, Dunkov BC, Dunn P, Durbin KJ, Evangelista CC, Ferraz C, Ferriera S, Fleischmann W, Fosler C, Gabrielian AE, Garg NS, Gelbart WM, Glasser K, Glodek A, Gong F, Gorrell JH, Gu Z, Guan P, Harris M, Harris NL, Harvey D, Heiman TJ, Hernandez JR, Houck J, Hostin D, Houston KA, Howland TJ, Wei MH, Ibegwam C, Jalali M, Kalush F, Karpen GH, Ke Z, Kennison JA, Ketchum KA, Kimmel BE, Kodira CD, Kraft C, Kravitz S, Kulp D, Lai Z, Lasko P, Lei Y, Levitsky AA, Li J, Li Z, Liang Y, Lin X, Liu X, Mattei B, McIntosh TC, McLeod MP, McPherson D, Merkulov G, Milshina NV, Mobarry C, Morris J, Moshrefi A, Mount SM, Moy M, Murphy B, Murphy L, Muzny DM, Nelson DL, Nelson DR, Nelson KA, Nixon K, Nusskern DR, Pacleb JM, Palazzolo M, Pittman GS, Pan S, Pollard J, Puri V, Reese MG, Reinert K, Remington K, Saunders RD, Scheeler F, Shen H, Shue BC, Sidén-Kiamos I, Simpson M, Skupski MP, Smith T, Spier E, Spradling AC, Stapleton M, Strong R, Sun E, Svirskas R, Tector C, Turner R, Venter E, Wang AH, Wang X, Wang ZY, Wassarman DA, Weinstock GM, Weissenbach J, Williams SM, WoodageT, Worley KC, Wu D, Yang S, Yao QA, Ye J, Yeh RF, Zaveri JS, Zhan M, Zhang G, Zhao Q, Zheng L, Zheng XH, Zhong FN, Zhong W, Zhou X, Zhu S, Zhu X, Smith HO, Gibbs RA, Myers EW, Rubin GM, Venter JC</div><div class="citation">Science. 2000 Mar 24; 287(5461):2185-95.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/10731132/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R6">Ref list</a>]</p></div><div id="body-link-popper-R7" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Whole-genome random sequencing and assembly of Haemophilus influenzae Rd.<div class="authors">Fleischmann RD, Adams MD, White O, Clayton RA, Kirkness EF, Kerlavage AR, Bult CJ, Tomb JF, Dougherty BA, Merrick JM</div><div class="citation">Science. 1995 Jul 28; 269(5223):496-512.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/7542800/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R7">Ref list</a>]</p></div><div id="body-link-popper-R12" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">12. <span class="element-citation">de Bruijn N. A combinatorial problem. <span><span class="ref-journal">Proc. Nederl. Akad. Wetensch. </span>1946;<span class="ref-vol">49</span>:758–764.</span></span> [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R12">Ref list</a>]</div><div id="body-link-popper-R13" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Sequencing of megabase plus DNA by hybridization: theory of the method.<div class="authors">Drmanac R, Labat I, Brukner I, Crkvenjakov R</div><div class="citation">Genomics. 1989 Feb; 4(2):114-28.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/2737674/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R13">Ref list</a>]</p></div><div id="body-link-popper-R15" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">[Determination of the nucleotide sequence of DNA using hybridization with oligonucleotides. A new method].<div class="authors">Lysov IuP, Florent'ev VL, Khorlin AA, Khrapko KR, Shik VV</div><div class="citation">Dokl Akad Nauk SSSR. 1988; 303(6):1508-11.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/3250844/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R15">Ref list</a>]</p></div><div id="body-link-popper-R16" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true"><span class="invert">Review</span> 1-Tuple DNA sequencing: computer analysis.<div class="authors">Pevzner PA</div><div class="citation">J Biomol Struct Dyn. 1989 Aug; 7(1):63-73.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/2684223/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R16">Ref list</a>]</p></div><div id="body-link-popper-R8" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Assembly of large genomes using second-generation sequencing.<div class="authors">Schatz MC, Delcher AL, Salzberg SL</div><div class="citation">Genome Res. 2010 Sep; 20(9):1165-73.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20508146/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R8">Ref list</a>]</p></div><div id="body-link-popper-R9" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Automated de novo protein sequencing of monoclonal antibodies.<div class="authors">Bandeira N, Pham V, Pevzner P, Arnott D, Lill JR</div><div class="citation">Nat Biotechnol. 2008 Dec; 26(12):1336-8.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/19060866/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R9">Ref list</a>]</p></div><div id="body-link-popper-R10" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">DRIMM-Synteny: decomposing genomes into evolutionary conserved segments.<div class="authors">Pham SK, Pevzner PA</div><div class="citation">Bioinformatics. 2010 Oct 15; 26(20):2509-16.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20736338/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R10">Ref list</a>]</p></div><div id="body-link-popper-R11" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Full-length transcriptome assembly from RNA-Seq data without a reference genome.<div class="authors">Grabherr MG, Haas BJ, Yassour M, Levin JZ, Thompson DA, Amit I, Adiconis X, Fan L, Raychowdhury R, Zeng Q, Chen Z, Mauceli E, Hacohen N, Gnirke A, Rhind N, di Palma F, Birren BW, Nusbaum C, Lindblad-Toh K, Friedman N, Regev A</div><div class="citation">Nat Biotechnol. 2011 May 15; 29(7):644-52.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/21572440/" ref="reftype=pubmed&amp;article-id=5531759&amp;issue-id=295986&amp;journal-id=319&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#R11">Ref list</a>]</p></div></div>
                        </div>
                        <div class="bottom">
                            
                            <div id="NCBIFooter_dynamic">
    <a id="help-desk-link" class="help_desk" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/" target="_blank">Support Center</a>
    <a id="help-desk-link" class="help_desk" href="https://support.ncbi.nlm.nih.gov/ics/support/KBList.asp?Time=2018-05-29T15:57:07-04:00&amp;Snapshot=%2Fprojects%2FPMC%2FPMCViewer@4.46&amp;Host=ptpmc201&amp;ncbi_phid=F4FB7F81B0DB00B100000000005B0054&amp;ncbi_session=F4FB3255AECCE8E1_0176SID&amp;from=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC5531759%2F&amp;Db=pmc&amp;folderID=132&amp;Ncbi_App=pmc&amp;Page=literature&amp;style=classic&amp;deptID=28049" target="_blank">Support Center</a>
    
</div>

                            <div class="footer" id="footer">
    
    <div class="subfooter"><!--?xml version="1.0" encoding="UTF-8"?-->
<div><h2 class="offscreen_noflow">Simple NCBI Directory</h2><ul class="foot_list" id="getting_started">
            <li><h3>Getting Started</h3></li>
            
            <li><a href="https://www.ncbi.nlm.nih.gov/home/learn.shtml">NCBI Education</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/books/NBK3831/">NCBI Help Manual</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/books/NBK143764/">NCBI Handbook</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/home/learn.shtml">Training &amp; Tutorials</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/home/submit.shtml">Submit Data</a></li>
        </ul><ul class="foot_list" id="resources">
            <li><h3>Resources</h3></li>
            
        <li><a href="https://www.ncbi.nlm.nih.gov/guide/chemicals-bioassays">Chemicals &amp; Bioassays</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/data-software">Data &amp; Software</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/dna-rna">DNA &amp; RNA</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/domains-structures">Domains &amp; Structures</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/genes-expression">Genes &amp; Expression</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/genetics-medicine">Genetics &amp; Medicine</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/genomes-maps">Genomes &amp; Maps</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/homology">Homology</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/literature">Literature</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/proteins">Proteins</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/sequence-analysis">Sequence Analysis</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/taxonomy">Taxonomy</a></li><li><a href="https://www.ncbi.nlm.nih.gov/guide/variation">Variation</a></li></ul><ul class="foot_list" id="popular">
            <li><h3>Popular</h3></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/">PubMed</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/books/">Bookshelf</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/pmc/">PubMed Central</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/pubmedhealth/">PubMed Health</a></li>
            <li><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="newdomain">BLAST</a></li>            
            <li><a href="https://www.ncbi.nlm.nih.gov/nucleotide/">Nucleotide</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/genome/">Genome</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/snp/">SNP</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/gene/">Gene</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/protein/">Protein</a></li>
            <li><a href="https://pubchem.ncbi.nlm.nih.gov/" class="newdomain">PubChem</a></li>
        </ul><ul class="foot_list" id="featured">
            <li><h3>Featured</h3></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/gtr/">Genetic Testing Registry</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/pubmedhealth/">PubMed Health</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/genbank/">GenBank</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/refseq/">Reference Sequences</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/geo/">Gene Expression Omnibus</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/genome/gdv/">Genome Data Viewer</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/genome/guide/human/">Human Genome</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/genome/guide/mouse/">Mouse Genome</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/genomes/FLU/">Influenza Virus</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/tools/primer-blast/">Primer-BLAST</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/Traces/sra/">Sequence Read Archive</a></li>
        </ul><ul class="foot_list" id="info">
            <li><h3>NCBI Information</h3></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/home/about/">About NCBI</a></li>
            <li><a href="https://www.ncbi.nlm.nih.gov/research/">Research at NCBI</a></li>
            <li><a href="https://ncbiinsights.ncbi.nlm.nih.gov/">NCBI News &amp; Blog</a></li>
            <li><a href="ftp://ftp.ncbi.nlm.nih.gov/">NCBI FTP Site</a></li>
            <li><a class="external" href="https://www.facebook.com/ncbi.nlm" target="_blank">NCBI on Facebook</a></li>
            <li><a class="external" href="https://twitter.com/ncbi" target="_blank">NCBI on Twitter</a></li>
            <li><a class="external" href="https://www.youtube.com/ncbinlm" target="_blank">NCBI on YouTube</a></li>
        </ul></div>
</div><script type="text/javascript" src="./Why are de Bruijn graphs useful for genome assembly__files/preloaderWidget.js"> </script>
    <div id="external-disclaimer" class="offscreen_noflow">
        External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.
    </div>    
    <div id="ncbifooter" class="contact_info">      
        <div id="footer-contents-right">
            <div id="nlm_thumb_logo">
                <a href="https://www.nlm.nih.gov/" title="NLM">NLM</a>
            </div>
            <div id="nih_thumb_logo">
                <a href="https://www.nih.gov/" title="NIH">NIH</a>
            </div>
            <div id="hhs_thumb_logo">
                <a href="https://www.hhs.gov/" title="DHHS">DHHS</a>
            </div>
            <div id="usagov_thumb_logo">
                <a href="https://www.usa.gov/" title="USA.gov">USA.gov</a>
            </div>         
        </div>
        
        <div id="footer-contents-left">
            <p class="address vcard">
                <span class="url">
                    <a class="fn url newdomain" href="https://www.ncbi.nlm.nih.gov/">National Center for
                        Biotechnology Information</a>,
                </span> <span class="org url newdomain"><a href="https://www.nlm.nih.gov/">U.S. National Library of Medicine</a></span>
                <span class="adr">
                    <span class="street-address">8600 Rockville Pike</span>, <span class="locality">Bethesda</span>
                    <span class="region">MD</span>, <span class="postal-code">20894</span>
                    <span class="country-name">USA</span>
                </span>
            </p>
            
            <a href="https://www.ncbi.nlm.nih.gov/home/about/policies.shtml">Policies and Guidelines</a> | <a href="https://www.ncbi.nlm.nih.gov/home/about/contact.shtml">Contact</a>
        </div>
    </div>
    <script type="text/javascript" src="./Why are de Bruijn graphs useful for genome assembly__files/InstrumentPageStarterJS.js"> </script>    
    <script type="text/javascript" src="./Why are de Bruijn graphs useful for genome assembly__files/hfjs2.js"> </script>
</div>
                        </div>
                    </div>
                    <!--/.page-->
                </div>
                <!--/.wrap-->
            </div><!-- /.twelve_col -->
        <div class="ui-ncbiautocomplete-holder shadow ui-ncbiautocomplete-holder-clearfix" aria-live="assertive" style="display: none; top: 63.5938px; left: 328.188px; width: 670px; z-index: 1001;"><ul class="ui-ncbiautocomplete-options" role="listbox" aria-activedescendant="term"></ul><div class="ui-ncbiautocomplete-actions shadow" style="display: block;"><a class="ui-ncbiautocomplete-link-pref" style="display: none;">Preferences</a><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/#" class="ui-ncbiautocomplete-link-off ui-ncbiautocomplete-link-pref-right" style="display: block;">Turn off</a></div></div><div style="display: none; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true" class="ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div><div style="display: none;">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div></div><script async="1" src="./Why are de Bruijn graphs useful for genome assembly__files/analytics.js"></script>
        <!-- /.grid -->

        <span class="PAFAppResources"></span>
        
        <!-- BESelector tab -->
        
        
        
        <noscript><img alt="statistics" src="/stat?jsdisabled=true&amp;ncbi_db=pmc&amp;ncbi_pdid=article&amp;ncbi_acc=&amp;ncbi_domain=nihpa&amp;ncbi_report=record&amp;ncbi_type=fulltext&amp;ncbi_objectid=&amp;ncbi_pcid=/articles/PMC5531759/&amp;ncbi_app=pmc" /></noscript>
        
        
        <!-- usually for JS scripts at page bottom -->
        <!--<component id="PageFixtures" label="styles"></component>-->
    

<!-- F4FB3255AECCE8E1_0176SID /projects/PMC/PMCViewer@4.46 ptpmc201 v4.1.r563004 Tue, May 01 2018 03:01:28 -->

<script type="text/javascript" async="" id="smcx-sdk" src="./Why are de Bruijn graphs useful for genome assembly__files/tRaiETqnLgj758hTBazgd_2Fzt773GyyxFLU9m0FUrhBjm_2BZjhMtZFwIJOzIRZkSuY.js"></script><script type="text/javascript" src="./Why are de Bruijn graphs useful for genome assembly__files/4065628.js" snapshot="pmc"></script>
<div class="ui-dialog ui-widget ui-widget-content ui-corner-all ui-front" tabindex="-1" role="dialog" aria-describedby="epubDialog" aria-labelledby="ui-id-1" style="display: none;"><div class="ui-dialog-titlebar ui-widget-header ui-corner-all ui-helper-clearfix"><span id="ui-id-1" class="ui-dialog-title">Making articles easier to read in PMC</span><button type="button" class="ui-button ui-widget ui-state-default ui-corner-all ui-button-icon-only ui-dialog-titlebar-close" role="button" title="Close"><span class="ui-button-icon-primary ui-icon ui-icon-closethick"></span><span class="ui-button-text">Close</span></button></div><div id="epubDialog" style="display: block;" class="ui-dialog-content ui-widget-content">  <p>We are experimenting with display styles that make it easier to read articles     in PMC.     Our first effort uses eBook readers, which have several "ease of reading"     features already built in.</p>  <p>These PMC articles are best viewed in the <em>iBooks     reader</em>. You may notice problems with the display of certain parts of an article     in other eReaders.</p>  <div class="ui-dialog-buttonpane ui-widget-content ui-helper-clearfix"><button id="cancelEpub" class="ui-state-default ui-corner-all">Cancel</button><button id="downloadEpub" style="float: left" class="ui-state-default ui-corner-all">Download article</button></div>          </div></div><ul class="print-log"><li></li></ul><div class="ui-ncbipopper-arrow-image ui-ncbipopper-arrow-right" style="display: block; top: 428.656px; left: 813.453px; opacity: 1;">&nbsp;</div></body></html>